{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Single sample analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Parsing the outputs with answers (BioData1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 413,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['#CHROM', 'POS', 'REF', 'ALT', ' Mappability ', 'VAF_Validation', 'Sample', 'Group', 'Validation']\n",
      "TRUE 117\n",
      "FALSE 13\n",
      "TRUE_VAF 117\n",
      "FALSE_VAF 13\n",
      "GBM-P-0396-Peri-M-Rep1:chr1:10659715:C:T\n"
     ]
    }
   ],
   "source": [
    "f_ans = open(\"./Answer.txt\", 'r')\n",
    "dic_ans = {\"TRUE\":[], 'FALSE':[], \"TRUE_VAF\" : [], \"FALSE_VAF\": []}\n",
    "\n",
    "#For set analysis\n",
    "l_TP_set = []\n",
    "l_FP_set = []\n",
    "\n",
    "\n",
    "for line in f_ans:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if line[0] == \"#\":\n",
    "        l_header = s\n",
    "        print (l_header)\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        #print (dic)\n",
    "        sample = dic[\"Sample\"]\n",
    "        chrom = dic[\"#CHROM\"]\n",
    "        pos = dic[\"POS\"]\n",
    "        ref = dic[\"REF\"]\n",
    "        alt = dic[\"ALT\"]\n",
    "        VAF = float(dic['VAF_Validation'].rstrip('%')) * 0.01\n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        shared_stat = None\n",
    "\n",
    "        validation = dic[\"Validation\"]\n",
    "        if validation == \"TRUE\":\n",
    "            dic_ans['TRUE'].append(info)\n",
    "            dic_ans['TRUE_VAF'].append(float(VAF))\n",
    "            l_TP_set.append(info)\n",
    "        else:\n",
    "            dic_ans['FALSE'].append(info)\n",
    "            dic_ans['FALSE_VAF'].append(float(VAF))\n",
    "            l_FP_set.append(info)\n",
    "\n",
    "for stat in dic_ans:\n",
    "    print (stat, len(dic_ans[stat]))\n",
    "    \n",
    "\n",
    "print (dic_ans[\"TRUE\"][0])\n",
    "#def Performance_gather(info, VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF):\n",
    "def Performance_gather(info, VAF, cnt_true, cnt_false,l_false, l_false_VAF):\n",
    "    if info in dic_ans[\"TRUE\"]:\n",
    "            cnt_true +=1\n",
    "            l_true.append(info)\n",
    "    elif info in dic_ans[\"FALSE\"]:\n",
    "            cnt_false +=1\n",
    "            l_false.append(info)\n",
    "            l_false_VAF.append(VAF)\n",
    "    else:\n",
    "        pass\n",
    "            #print (info, \"Neither in TRUE nor FALSE\")\n",
    "    return [cnt_true, cnt_false, l_false, l_false_VAF]\n",
    "\n",
    "from scipy.stats import hmean\n",
    "def Performance_check(cnt_true, cnt_false):\n",
    "    TP = cnt_true\n",
    "    FP = cnt_false\n",
    "    PC = len(dic_ans['TRUE'])\n",
    "    NC = len(dic_ans['FALSE'])\n",
    "    sens = TP/PC\n",
    "    if TP == 0 and FP== 0:\n",
    "        pre = None\n",
    "        f_score = None\n",
    "    else:\n",
    "        pre = TP/(TP + FP)\n",
    "        f_score = hmean([sens,pre])\n",
    "    print (sens, pre, f_score)\n",
    "    return (TP, FP, sens, pre, f_score)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 401,
   "metadata": {},
   "outputs": [],
   "source": [
    "DATA = \"Kim et al.\"\n",
    "VAF_range = \"Total\"\n",
    "l_result = []\n",
    "out = open('/data/project/MRS/BioValidation/PlosGen/analysis/Parse_result.txt', 'w')\n",
    "out.write(\"%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n\" %(\"tool\", \"TP\", \"FP\", \"Sensitivity\", \"Precision\", \"F1-score\", \"Data\", \"VAF_range\"))\n",
    "out_set_TP = open('/data/project/MRS/BioValidation/PlosGen/analysis/Parse_set_TP.txt', 'w')\n",
    "out_set_FP = open('/data/project/MRS/BioValidation/PlosGen/analysis/Parse_set_FP.txt', 'w')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 402,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.6324786324786325 0.9367088607594937 0.7551020408163265\n",
      "(74, 5, 0.6324786324786325, 0.9367088607594937, 0.7551020408163265)\n",
      "['AD-P-2013029-HIF-F-Rep1:chr8:123182166:A:G', 'AD-P-2013029-HIF-F-Rep1:chr22:40315361:G:A', 'AD-P-2013066-HIF-F-Rep1:chr4:99347143:A:G', 'AD-P-2013066-HIF-F-Rep1:chr13:42992009:G:A', 'AD-P-2013066-HIF-F-Rep1:chr15:98896836:A:T', 'AD-P-2013066-HIF-F-Rep1:chr20:17494172:C:T', 'AD-P-2013066-HIF-F-Rep1:chr22:46257097:G:A', 'AD-P-2014054-HIF-F-Rep1:chr11:67628329:C:T', 'ASD-P-1349-Occi-M-Rep1:chr12:4626950:G:T', 'ASD-P-1638-PFC-F-Rep1:chr12:113191328:G:A', 'ASD-P-4899-Occi-M-Rep1:chr5:179559872:T:A', 'ASD-P-5574-Heart-M-Rep1:chr15:74722831:C:T', 'ASD-P-5574-Occi-M-Rep1:chr9:84294263:C:T', 'ASD-P-5574-Occi-M-Rep1:chr15:74722831:C:T', 'FCD-P-0021-Blood-M-Rep1:chr16:69463844:A:G', 'FCD-P-0021-RTemOcci-M-Rep1:chr16:69463844:A:G', 'FCD-P-0023-Blood-M-Rep1:chr20:36209646:G:A', 'FCD-P-0023-Lhem-M-Rep1:chr20:36209646:G:A', 'GBM-P-0026-Tumor-F-Rep2:chr11:128480240:G:A', 'GBM-P-0026-Tumor-F-Rep2:chr21:33262947:C:A', 'GBM-P-0027-Tumor-M-Rep1:chr6:87245535:A:G', 'GBM-P-0027-Tumor-M-Rep1:chr17:74352100:C:T', 'GBM-P-0052-SVZ-M-Rep2:chr17:9969721:C:G', 'GBM-P-0052-Tumor-M-Rep1:chr12:89425318:A:T', 'GBM-P-0160-Tumor-M-Rep1:chr7:49983464:C:T', 'GBM-P-0160-Tumor-M-Rep1:chr15:41844534:C:T', 'GBM-P-0160-Tumor-M-Rep1:chr15:99716488:G:A', 'GBM-P-0187-Blood-M-Rep1:chr9:127676696:G:A', 'GBM-P-0187-SVZ-M-Rep1:chr9:127676696:G:A', 'GBM-P-0187-Tumor-M-Rep1:chr2:210433263:T:C', 'GBM-P-0187-Tumor-M-Rep1:chr5:113434693:G:A', 'GBM-P-0276-Tumor-M-Rep1:chr1:52337866:G:A', 'GBM-P-0276-Tumor-M-Rep1:chr2:119248056:C:A', 'GBM-P-0276-Tumor-M-Rep1:chr4:86001514:C:T', 'GBM-P-0276-Tumor-M-Rep1:chr6:158095820:G:A', 'GBM-P-0276-Tumor-M-Rep1:chr11:59422881:T:C', 'GBM-P-0276-Tumor-M-Rep1:chr13:78600306:C:G', 'GBM-P-0276-Tumor-M-Rep1:chr18:31469150:G:A', 'GBM-P-0276-Tumor-M-Rep1:chr21:45909407:C:T', 'GBM-P-0396-Peri-M-Rep1:chr1:10659715:C:T', 'GBM-P-0396-Recur-M-Rep1:chr1:10659715:C:T', 'GBM-P-0396-Recur-M-Rep1:chr1:116031514:G:A', 'GBM-P-0396-Recur-M-Rep1:chr2:238271458:T:C', 'GBM-P-0396-Recur-M-Rep1:chr4:40103074:T:C', 'GBM-P-0396-Recur-M-Rep1:chr7:70791458:T:C', 'GBM-P-0396-Recur-M-Rep1:chr20:59994312:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr2:24039486:C:A', 'GBM-P-0881-Tumor-M-Rep1:chr2:218164082:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr3:197090990:C:T', 'GBM-P-0881-Tumor-M-Rep1:chr11:66332605:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr11:129864369:T:C', 'GBM-P-0881-Tumor-M-Rep1:chr15:94479623:A:T', 'GBM-P-0881-Tumor-M-Rep1:chr16:54112977:G:T', 'GBM-P-0881-Tumor-M-Rep1:chr17:7675094:A:C', 'GBM-P-0881-Tumor-M-Rep1:chr17:32489134:C:A', 'GG-P-0231-Uncus-M-Rep1:chr5:123180244:G:A', 'NLE-P-0225-Brain-M-Rep1:chr1:198742295:C:T', 'NLE-P-0225-Brain-M-Rep1:chr1:201047600:G:A', 'NLE-P-0225-Brain-M-Rep1:chr1:204450011:G:C', 'NLE-P-0225-Brain-M-Rep1:chr1:229595008:G:A', 'NLE-P-0225-Brain-M-Rep1:chr1:236569023:C:A', 'NLE-P-0227-Brain-M-Rep1:chr19:6583304:C:T', 'SCZ-C-0002-Liver-M-Rep1:chr2:219420155:G:A', 'SCZ-C-0002-Liver-M-Rep1:chr6:136192149:C:T', 'SCZ-C-0002-PFC-M-Rep1:chr2:219420155:G:A', 'SCZ-C-0002-PFC-M-Rep1:chr6:136192149:C:T', 'SCZ-C-1021-Liver-M-Rep1:chr16:27749710:C:T', 'SCZ-C-1021-PFC-M-Rep2:chr16:27749710:C:T', 'SCZ-P-0021-PFC-M-Rep1:chr15:73998214:C:T', 'SCZ-P-1002-Liver-M-Rep1:chr22:38670902:G:A', 'SCZ-P-1002-PFC-M-Rep1:chr22:38670902:G:A', 'SCZ-P-1004-PFC-M-Rep1:chr12:54966142:G:A', 'AD-P-2013066-HIF-F-Rep1:chrX:107576369:G:A', 'GBM-P-0160-Tumor-M-Rep1:chrX:154860535:C:T']\n",
      "['ASD-P-4721-Temp-M-Rep1:chr20:46366499:G:A', 'GBM-P-0026-Tumor-F-Rep2:chr19:8371354:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T', 'SCZ-C-1014-PFC-M-Rep2:chr2:138015377:A:G', 'SCZ-P-1001-PFC-M-Rep1:chr13:112558343:G:A']\n",
      "['0.0373333333333333', '0.0607142857142857', '0.0835913312693498', '0.0459770114942529', '0.0317460317460317']\n",
      "0.6324786324786325 0.9367088607594937 0.7551020408163265\n",
      "['ASD-P-4721-Temp-M-Rep1:chr20:46366499:G:A', 'GBM-P-0026-Tumor-F-Rep2:chr19:8371354:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T', 'SCZ-C-1014-PFC-M-Rep2:chr2:138015377:A:G', 'SCZ-P-1001-PFC-M-Rep1:chr13:112558343:G:A']\n",
      "['1', '1', '0', '0', '0', '0', '0', '0', '1', '1', '0', '1', '0']\n"
     ]
    }
   ],
   "source": [
    "#MF\n",
    "f = open('/data/project/MRS/BioValidation/PlosGen/output/MF/All.snv.Predictions.Mosaic', 'r')\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "\n",
    "for line in f:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if s[0] == \"id\":\n",
    "        l_header = s\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        info_tmp = dic['id']\n",
    "        VAF = dic['AF']\n",
    "        info_tmp = info_tmp.split('/')[-1]\n",
    "        #print (info_tmp.split('~'))\n",
    "        info = ':'.join(info_tmp.split('~'))\n",
    "        #print (info) #AD-P-2013029-HIF-F-Rep1:chr8:123182166:A:G\n",
    "        cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "        cnt_true = cnts[0]\n",
    "        cnt_false = cnts[1]\n",
    "        l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_true)\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "\n",
    "\n",
    "l_result_tmp = ['MF']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n",
    "l_TP_decide = []\n",
    "l_FP_decide = []\n",
    "for locus in l_TP_set:\n",
    "    if locus in l_true:\n",
    "        l_TP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_TP_decide.append(\"0\")\n",
    "        \n",
    "for locus in l_FP_set:\n",
    "    if locus in l_false:\n",
    "        l_FP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_FP_decide.append(\"0\")\n",
    "\n",
    "out_set_TP.write(\"MF\" + '\\t' + '\\t'.join(l_TP_decide) + '\\n')\n",
    "out_set_FP.write(\"MF\" + '\\t' + '\\t'.join(l_FP_decide) + '\\n')\n",
    "\n",
    "print (l_false)\n",
    "print (l_FP_decide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 403,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "0.8547008547008547 0.9259259259259259 0.8888888888888888\n",
      "(100, 8, 0.8547008547008547, 0.9259259259259259, 0.8888888888888888)\n",
      "['GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A', 'SCZ-P-1001-PFC-M-Rep1:chr13:112558343:G:A', 'NLE-P-0228-Occi-F-Rep1:chr7:100964707:C:T', 'GBM-P-0026-Tumor-F-Rep2:chr19:8371354:G:A', 'SCZ-C-1014-PFC-M-Rep2:chr2:138015377:A:G', 'ASD-P-4721-Temp-M-Rep1:chr20:46366499:G:A', 'NLE-P-0041-Frontal-F-Rep1:chr21:30425715:A:G', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T']\n",
      "['0.279', '0.040', '0.247', '0.069', '0.057', '0.040', '0.037', '0.085']\n",
      "0.8547008547008547 0.9259259259259259 0.8888888888888888\n",
      "['1', '1', '1', '1', '0', '0', '0', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '0', '0', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '0', '0', '0', '0', '0', '0', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '0', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '1', '0', '1', '1', '0', '0', '1', '0', '1', '1', '1', '0', '1', '1', '1']\n"
     ]
    }
   ],
   "source": [
    "#MT2\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/MT2/*PASS.vcf\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            print (s[8])\n",
    "            VAF = s[9].split(':')[2]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #ref = float(allele.split(',')[0])\n",
    "            #alt = float(allele.split(',')[0])\n",
    "            #VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['MT2']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n",
    "l_TP_decide = []\n",
    "l_FP_decide = []\n",
    "for locus in l_TP_set:\n",
    "    if locus in l_true:\n",
    "        l_TP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_TP_decide.append(\"0\")\n",
    "for locus in l_FP_set:\n",
    "    if locus in l_false:\n",
    "        l_FP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_FP_decide.append(\"0\")\n",
    "\n",
    "out_set_TP.write(\"MT2\" + '\\t' + '\\t'.join(l_TP_decide) + '\\n')\n",
    "out_set_FP.write(\"MT2\" + '\\t' + '\\t'.join(l_FP_decide) + '\\n')\n",
    "\n",
    "print (l_TP_decide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 404,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "15 2\n",
      "0.1282051282051282 0.8823529411764706 0.22388059701492538\n",
      "(15, 2, 0.1282051282051282, 0.8823529411764706, 0.22388059701492538)\n",
      "['SCZ-C-1014-PFC-M-Rep2:chr2:138015377:A:G', 'SCZ-P-1001-PFC-M-Rep1:chr13:112558343:G:A']\n",
      "['0.048', '0.02564102564102564']\n",
      "0.1282051282051282 0.8823529411764706 0.22388059701492538\n",
      "['0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '1', '0', '0', '0', '0', '0', '1', '1', '1', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '1', '1', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '1', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0']\n"
     ]
    }
   ],
   "source": [
    "#DM\n",
    "f = open('/data/project/MRS/BioValidation/PlosGen/output/DM_output/All.DM.output.mosaic.txt', 'r')\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "\n",
    "for line in f:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    sample = s[0]\n",
    "    chrom = s[2]\n",
    "    pos = s[3]\n",
    "    ref = s[4]\n",
    "    alt = s[5]\n",
    "    VAF = s[7]\n",
    "    info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "\n",
    "    cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "    cnt_true = cnts[0]\n",
    "    cnt_false = cnts[1]\n",
    "    l_false = cnts[2]\n",
    "    \n",
    "print (cnt_true, cnt_false)\n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['DM']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n",
    "l_TP_decide = []\n",
    "l_FP_decide = []\n",
    "for locus in l_TP_set:\n",
    "    if locus in l_true:\n",
    "        l_TP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_TP_decide.append(\"0\")\n",
    "for locus in l_FP_set:\n",
    "    if locus in l_false:\n",
    "        l_FP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_FP_decide.append(\"0\")\n",
    "\n",
    "out_set_TP.write(\"DM\" + '\\t' + '\\t'.join(l_TP_decide) + '\\n')\n",
    "out_set_FP.write(\"DM\" + '\\t' + '\\t'.join(l_FP_decide) + '\\n')\n",
    "\n",
    "print (l_TP_decide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 405,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.3418803418803419 0.9523809523809523 0.5031446540880503\n",
      "(40, 2, 0.3418803418803419, 0.9523809523809523, 0.5031446540880503)\n",
      "['NLE-P-0228-Occi-F-Rep1:chr7:100964707:C:T', 'GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A']\n",
      "[0.245662100456621, 0.27673649393605293]\n",
      "0.3418803418803419 0.9523809523809523 0.5031446540880503\n",
      "['1', '1', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '1', '1', '0', '0', '1', '1', '1', '0', '1', '1', '0', '0', '0', '0', '0', '0', '0', '1', '1', '0', '0', '0', '0', '1', '1', '1', '0', '0', '0', '1', '0', '0', '1', '1', '1', '1', '0', '1', '1', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '1', '1', '0', '0', '0', '1', '0', '1', '0', '0', '1', '0', '0', '1', '0', '1', '0', '0', '1', '1', '0', '1', '0', '0', '0', '0', '1', '1', '0', '0', '1', '0', '0', '0', '0', '1', '0', '0', '1']\n"
     ]
    }
   ],
   "source": [
    "#HCp20\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/HCp20/*PASS.vcf\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            allele = s[9].split(':')[1]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            ref = float(allele.split(',')[0])\n",
    "            alt = float(allele.split(',')[1])\n",
    "            VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['HC-p20']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n",
    "l_TP_decide = []\n",
    "l_FP_decide = []\n",
    "for locus in l_TP_set:\n",
    "    if locus in l_true:\n",
    "        l_TP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_TP_decide.append(\"0\")\n",
    "for locus in l_FP_set:\n",
    "    if locus in l_false:\n",
    "        l_FP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_FP_decide.append(\"0\")\n",
    "\n",
    "out_set_TP.write(\"HC-p20\" + '\\t' + '\\t'.join(l_TP_decide) + '\\n')\n",
    "out_set_FP.write(\"HC-p20\" + '\\t' + '\\t'.join(l_FP_decide) + '\\n')\n",
    "\n",
    "print (l_TP_decide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 406,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.38461538461538464 0.9375 0.5454545454545455\n",
      "(45, 3, 0.38461538461538464, 0.9375, 0.5454545454545455)\n",
      "['GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T', 'NLE-P-0228-Occi-F-Rep1:chr7:100964707:C:T', 'GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A']\n",
      "[0.08284023668639054, 0.245662100456621, 0.27673649393605293]\n",
      "0.38461538461538464 0.9375 0.5454545454545455\n",
      "['1', '1', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '1', '1', '0', '0', '1', '1', '1', '0', '1', '1', '0', '0', '0', '0', '0', '0', '0', '1', '1', '0', '0', '0', '0', '1', '1', '1', '1', '0', '0', '1', '0', '0', '1', '1', '1', '1', '0', '1', '1', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '1', '0', '1', '1', '0', '0', '0', '1', '0', '1', '0', '0', '1', '0', '0', '1', '0', '1', '1', '0', '1', '1', '0', '1', '0', '1', '0', '0', '1', '1', '0', '0', '1', '0', '0', '0', '1', '1', '0', '0', '1']\n"
     ]
    }
   ],
   "source": [
    "#HCp200\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/HCp200/*PASS.vcf\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            allele = s[9].split(':')[1]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            ref = float(allele.split(',')[0])\n",
    "            alt = float(allele.split(',')[1])\n",
    "            VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['HC-p200']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n",
    "\n",
    "l_TP_decide = []\n",
    "l_FP_decide = []\n",
    "for locus in l_TP_set:\n",
    "    if locus in l_true:\n",
    "        l_TP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_TP_decide.append(\"0\")\n",
    "for locus in l_FP_set:\n",
    "    if locus in l_false:\n",
    "        l_FP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_FP_decide.append(\"0\")\n",
    "\n",
    "out_set_TP.write(\"HC-p200\" + '\\t' + '\\t'.join(l_TP_decide) + '\\n')\n",
    "out_set_FP.write(\"HC-p200\" + '\\t' + '\\t'.join(l_FP_decide) + '\\n')\n",
    "\n",
    "print (l_TP_decide)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 407,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.1111111111111111 0.8666666666666667 0.196969696969697\n",
      "(13, 2, 0.1111111111111111, 0.8666666666666667, 0.196969696969697)\n",
      "['NLE-P-0228-Occi-F-Rep1:chr7:100964707:C:T', 'SCZ-C-1014-PFC-M-Rep2:chr2:138015377:A:G']\n",
      "[0.32675709001233044, 0.05309734513274336]\n",
      "0.1111111111111111 0.8666666666666667 0.196969696969697\n",
      "['0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '1', '0', '1', '0', '0', '1', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '1', '0', '0', '0', '0', '0', '0', '0', '0', '1', '1', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0', '0']\n"
     ]
    }
   ],
   "source": [
    "#MH\n",
    "import glob\n",
    "fl = glob.glob('/data/project/MRS/BioValidation/PlosGen/output/MH/*/run/final.passed.tsv')\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    sample = str(file).split('/')[-3]\n",
    "    for line in f:\n",
    "        s = line.strip().split('\\t')\n",
    "        chrom = s[0]\n",
    "        pos = s[1]\n",
    "        ref = s[6]\n",
    "        alt = s[8]\n",
    "        VAF = float(s[9])/int(s[7])\n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "        cnt_true = cnts[0]\n",
    "        cnt_false = cnts[1]\n",
    "        l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "        \n",
    "l_result_tmp = ['MH']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp )) + '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n",
    "\n",
    "l_TP_decide = []\n",
    "l_FP_decide = []\n",
    "for locus in l_TP_set:\n",
    "    if locus in l_true:\n",
    "        l_TP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_TP_decide.append(\"0\")\n",
    "for locus in l_FP_set:\n",
    "    if locus in l_false:\n",
    "        l_FP_decide.append(\"1\")\n",
    "    else:\n",
    "        l_FP_decide.append(\"0\")\n",
    "\n",
    "out_set_TP.write(\"MH\" + '\\t' + '\\t'.join(l_TP_decide) + '\\n')\n",
    "out_set_FP.write(\"MH\" + '\\t' + '\\t'.join(l_FP_decide) + '\\n')\n",
    "\n",
    "print (l_TP_decide)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Feature-level recombination"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['#CHROM', 'POS', 'REF', 'ALT', ' Mappability ', 'VAF_Validation', 'Sample', 'Group', 'Validation']\n",
      "TRUE 117\n",
      "FALSE 13\n",
      "TRUE_VAF 117\n",
      "FALSE_VAF 13\n",
      "GBM-P-0396-Peri-M-Rep1:chr1:10659715:C:T\n"
     ]
    }
   ],
   "source": [
    "f_ans = open(\"../../resource/Answer.txt\", 'r')\n",
    "dic_ans = {\"TRUE\":[], 'FALSE':[], \"TRUE_VAF\" : [], \"FALSE_VAF\": []}\n",
    "for line in f_ans:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if line[0] == \"#\":\n",
    "        l_header = s\n",
    "        print (l_header)\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        #print (dic)\n",
    "        sample = dic[\"Sample\"]\n",
    "        chrom = dic[\"#CHROM\"]\n",
    "        pos = dic[\"POS\"]\n",
    "        ref = dic[\"REF\"]\n",
    "        alt = dic[\"ALT\"]\n",
    "        VAF = float(dic['VAF_Validation'].rstrip('%')) * 0.01\n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        validation = dic[\"Validation\"]\n",
    "        if validation == \"TRUE\":\n",
    "            dic_ans['TRUE'].append(info)\n",
    "            dic_ans['TRUE_VAF'].append(float(VAF))\n",
    "        else:\n",
    "            dic_ans['FALSE'].append(info)\n",
    "            dic_ans['FALSE_VAF'].append(float(VAF))\n",
    "\n",
    "for stat in dic_ans:\n",
    "    print (stat, len(dic_ans[stat]))\n",
    "    \n",
    "\n",
    "\n",
    "print (dic_ans[\"TRUE\"][0])\n",
    "def Performance_gather(info, VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF):\n",
    "    if info in dic_ans[\"TRUE\"]:\n",
    "            cnt_true +=1\n",
    "            l_true.append(info)\n",
    "    elif info in dic_ans[\"FALSE\"]:\n",
    "            cnt_false +=1\n",
    "            l_false.append(info)\n",
    "            l_false_VAF.append(VAF)\n",
    "    else:\n",
    "        pass\n",
    "            #print (info, \"Neither in TRUE nor FALSE\")\n",
    "    return [cnt_true, cnt_false, l_false, l_false_VAF]\n",
    "\n",
    "from scipy.stats import hmean\n",
    "def Performance_check(cnt_true, cnt_false):\n",
    "    TP = cnt_true\n",
    "    FP = cnt_false\n",
    "    PC = len(dic_ans['TRUE'])\n",
    "    NC = len(dic_ans['FALSE'])\n",
    "    sens = TP/PC\n",
    "    if TP == 0 and FP== 0:\n",
    "        pre = None\n",
    "        f_score = None\n",
    "    else:\n",
    "        pre = TP/(TP + FP)\n",
    "        f_score = hmean([sens,pre])\n",
    "    print (sens, pre, f_score)\n",
    "    return (TP, FP, sens, pre, f_score)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## 1. MT2-to call set + alt softclip (MF) > 0.05\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.8547008547008547 0.9259259259259259 0.8888888888888888\n",
      "(100, 8, 0.8547008547008547, 0.9259259259259259, 0.8888888888888888)\n",
      "['GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A', 'SCZ-P-1001-PFC-M-Rep1:chr13:112558343:G:A', 'NLE-P-0228-Occi-F-Rep1:chr7:100964707:C:T', 'GBM-P-0026-Tumor-F-Rep2:chr19:8371354:G:A', 'SCZ-C-1014-PFC-M-Rep2:chr2:138015377:A:G', 'ASD-P-4721-Temp-M-Rep1:chr20:46366499:G:A', 'NLE-P-0041-Frontal-F-Rep1:chr21:30425715:A:G', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T']\n",
      "['0.279', '0.040', '0.247', '0.069', '0.057', '0.040', '0.037', '0.085']\n",
      "GBM-P-0187-Tumor-M-Rep1:chr20:17456366\n",
      "SCZ-P-1001-PFC-M-Rep1:chr13:112558343\n",
      "NLE-P-0228-Occi-F-Rep1:chr7:100964707\n",
      "GBM-P-0026-Tumor-F-Rep2:chr19:8371354\n",
      "SCZ-C-1014-PFC-M-Rep2:chr2:138015377\n",
      "ASD-P-4721-Temp-M-Rep1:chr20:46366499\n",
      "NLE-P-0041-Frontal-F-Rep1:chr21:30425715\n",
      "GBM-P-0881-Tumor-M-Rep1:chr1:110340663\n"
     ]
    }
   ],
   "source": [
    "#MT2\n",
    "import glob\n",
    "\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/MT2/*PASS.vcf\") \n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            #print (s[8])\n",
    "            VAF = s[9].split(':')[2]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #ref = float(allele.split(',')[0])\n",
    "            #alt = float(allele.split(',')[0])\n",
    "            #VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]     \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_MT_false = []\n",
    "for e_f in l_false:\n",
    "    s = e_f.split(':')\n",
    "    pos = ':'.join(s[0:3])\n",
    "    print (pos)\n",
    "    l_MT_false.append(pos)\n",
    "    \n",
    "l_MT_true = []\n",
    "for e_f in l_true:\n",
    "    s = e_f.split(':')\n",
    "    pos = ':'.join(s[0:3])\n",
    "    #print (pos)\n",
    "    l_MT_true.append(pos) \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FILTERED TP:  GBM-P-0881-Tumor-M-Rep1:chr2:24039486\n",
      "ORI TP:  100\n",
      "Filtered TP: 1\n",
      "ORI FP:  8\n",
      "Filtered FP:  0\n"
     ]
    }
   ],
   "source": [
    "f_MF_raw = open(\"/data/project/MRS/BioValidation/PlosGen/output/MF/All.snv.Predictions\", 'r')\n",
    "\n",
    "ori_fp = len(l_MT_false)\n",
    "ori_tp = len(l_MT_true)\n",
    "cnt_fil_tp, cnt_fil_fp = 0,0\n",
    "\n",
    "for line in f_MF_raw:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if s[0] == \"id\":\n",
    "        l_header = s\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        info_tmp = dic['id']\n",
    "        VAF = dic['AF']\n",
    "        info_tmp = info_tmp.lstrip('./')\n",
    "        info_tmp = info_tmp.split('~')[0:3]\n",
    "        #print (info_tmp.split('~'))\n",
    "        pos = ':'.join(info_tmp)\n",
    "        if pos in l_MT_false:\n",
    "            #print (pos, dic['id'])\n",
    "            alt_softclip = dic['alt_softclip']\n",
    "            if float(alt_softclip) > 0.05:\n",
    "                cnt_fil_fp +=1 \n",
    "        if pos in l_MT_true:\n",
    "            alt_softclip = dic['alt_softclip']\n",
    "            if float(alt_softclip) > 0.05:\n",
    "                cnt_fil_tp +=1 \n",
    "                print (\"FILTERED TP: \", pos)\n",
    "print (\"ORI TP: \", ori_tp)\n",
    "print (\"Filtered TP:\", cnt_fil_tp)\n",
    "print (\"ORI FP: \", ori_fp)\n",
    "print (\"Filtered FP: \", cnt_fil_fp)"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "## 2. HC-p200 call set + MFRL alt (MT2-to) <150\n",
    "## 3. HC-p200 call set + Het likelihood (MF) > 0.25"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.38461538461538464 0.9375 0.5454545454545455\n",
      "(45, 3, 0.38461538461538464, 0.9375, 0.5454545454545455)\n",
      "['GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T', 'NLE-P-0228-Occi-F-Rep1:chr7:100964707:C:T', 'GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A']\n",
      "[0.08284023668639054, 0.245662100456621, 0.27673649393605293]\n",
      "GBM-P-0881-Tumor-M-Rep1:chr1:110340663\n",
      "NLE-P-0228-Occi-F-Rep1:chr7:100964707\n",
      "GBM-P-0187-Tumor-M-Rep1:chr20:17456366\n"
     ]
    }
   ],
   "source": [
    "#HCp200\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/HCp200/*PASS.vcf\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "ori_fp = len(l_HC200_false)\n",
    "ori_tp = len(l_HC200_true)\n",
    "cnt_fil_tp, cnt_fil_fp = 0,0\n",
    "\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            allele = s[9].split(':')[1]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            ref = float(allele.split(',')[0])\n",
    "            alt = float(allele.split(',')[1])\n",
    "            VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_HC200_false = []\n",
    "for e_f in l_false:\n",
    "    s = e_f.split(':')\n",
    "    pos = ':'.join(s[0:3])\n",
    "    print (pos)\n",
    "    l_HC200_false.append(pos)\n",
    "    \n",
    "l_HC200_true = []\n",
    "for e_f in l_true:\n",
    "    s = e_f.split(':')\n",
    "    pos = ':'.join(s[0:3])\n",
    "    #print (pos)\n",
    "    l_HC200_true.append(pos) "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "FILTERED TP:  10659715 137\n",
      "FILTERED TP:  238271458 146\n",
      "FILTERED TP:  59994312 145\n",
      "FILTERED TP:  35803613 134\n",
      "FILTERED TP:  18781163 134\n",
      "FILTERED TP:  10659715 134\n",
      "ORI TP:  100\n",
      "Filtered TP: 6\n",
      "ORI FP:  8\n",
      "Filtered FP:  0\n"
     ]
    }
   ],
   "source": [
    "#HC200 with MT2-to MFRL alt < 150\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/MT2/*filtered.vcf\")\n",
    "ori_fp = len(l_MT_false)\n",
    "ori_tp = len(l_MT_true)\n",
    "cnt_fil_tp, cnt_fil_fp = 0,0\n",
    "\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            VAF = s[9].split(':')[2]\n",
    "            info = sample + ':' + chrom + ':' + pos\n",
    "            if info in l_HC200_false:\n",
    "                s_INFO = s[7].split(';')\n",
    "                for i in s_INFO:\n",
    "                    if \"MFRL\" in i:\n",
    "                        MFRL_alt = i.lstrip(\"MFRL=\").split(',')[1]\n",
    "                        if int(MFRL_alt) < 150:\n",
    "                            cnt_fil_fp +=1\n",
    "            if info in l_HC200_true:\n",
    "                s_INFO = s[7].split(';')\n",
    "                for i in s_INFO:\n",
    "                    if \"MFRL\" in i:\n",
    "                        MFRL_alt = i.lstrip(\"MFRL=\").split(',')[1]\n",
    "                        if int(MFRL_alt) < 150:\n",
    "                            cnt_fil_tp +=1\n",
    "                            print (\"FILTERED TP: \", pos, MFRL_alt)\n",
    "                            \n",
    "print (\"ORI TP: \", ori_tp)\n",
    "print (\"Filtered TP:\", cnt_fil_tp)\n",
    "print (\"ORI FP: \", ori_fp)\n",
    "print (\"Filtered FP: \", cnt_fil_fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "ORI TP:  100\n",
      "Filtered TP: 0\n",
      "ORI FP:  8\n",
      "Filtered FP:  0\n"
     ]
    }
   ],
   "source": [
    "#“Het likelihood” of MF (>0.25, 95% of adjustable true and 8% of germline false positives).\n",
    "\n",
    "f_MF_raw = open(\"/data/project/MRS/BioValidation/PlosGen/output/MF/All.snv.Predictions\", 'r')\n",
    "ori_fp = len(l_MT_false)\n",
    "ori_tp = len(l_MT_true)\n",
    "cnt_fil_tp, cnt_fil_fp = 0,0\n",
    "for line in f_MF_raw:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if s[0] == \"id\":\n",
    "        l_header = s\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        info_tmp = dic['id']\n",
    "        VAF = dic['AF']\n",
    "        info_tmp = info_tmp.lstrip('./')\n",
    "        info_tmp = info_tmp.split('~')[0:3]\n",
    "        #print (info_tmp.split('~'))\n",
    "        pos = ':'.join(info_tmp)\n",
    "        if pos in l_HC200_false:\n",
    "            #print (pos)\n",
    "            het_likelihood = float(dic['het_likelihood'])\n",
    "            if het_likelihood > 0.25:\n",
    "                cnt_fil_fp +=1\n",
    "        if pos in l_HC200_true:\n",
    "            het_likelihood = float(dic['het_likelihood'])\n",
    "            if het_likelihood > 0.25:\n",
    "                print (\"Filtered TP POS:\", pos)\n",
    "                cnt_fil_tp +=1\n",
    "\n",
    "print (\"ORI TP: \", ori_tp)\n",
    "print (\"Filtered TP:\", cnt_fil_tp)\n",
    "print (\"ORI FP: \", ori_fp)\n",
    "print (\"Filtered FP: \", cnt_fil_fp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Paired sample analysis"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Shared variants"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['X.CHROM', 'POS', 'REF', 'ALT', 'Mappability', 'VAF_Validation', 'Sample', 'Group', 'Validation', 'VAF', 'SET', 'info', 'n']\n",
      "TRUE 51\n",
      "FALSE 5\n",
      "TRUE_VAF 51\n",
      "FALSE_VAF 5\n",
      "DIC_SHARED2 {'TRUE': {'chr1:10659715': [22.07, 21.83], 'chr1:121183580': [8.5, 8.5], 'chr1:152308217': [2.6, 2.9, 2.9, 2.5], 'chr2:111858311': [3.6000000000000005, 3.6000000000000005], 'chr2:219420155': [4.2, 4.2], 'chr5:733811': [6.5, 6.4, 6.4, 6.800000000000001], 'chr5:837373': [5.4, 5.4], 'chr6:136192149': [7.000000000000001, 7.000000000000001], 'chr7:23585592': [24.5, 28.4], 'chr7:55157725': [2.5, 2.5], 'chr9:127676696': [1.8000000000000003, 1.7000000000000002], 'chr12:389107': [8.5, 8.0, 7.6, 8.1, 7.6, 7.8], 'chr12:122701415': [8.4, 8.4], 'chr12:122701470': [8.1, 9.5, 8.1], 'chr15:74722831': [2.3, 2.3], 'chr16:27749710': [7.400000000000001, 7.400000000000001], 'chr16:69463844': [3.3000000000000003, 3.3000000000000003], 'chr17:18781163': [10.6, 10.4], 'chr20:36209646': [4.5, 4.5], 'chr21:44550435': [7.400000000000001, 9.7], 'chr22:38670902': [4.0, 4.0]}, 'FALSE': {'chr5:173609431': [1.8000000000000003, 1.8000000000000003, 1.8000000000000003], 'chr12:103985722': [5.4, 5.4]}}\n",
      "GBM-P-0396-Peri-M-Rep1:chr1:10659715:C:T\n",
      "ASD-C-5408-Occi-M-Rep1:chr5:173609431:T:A\n",
      "ASD-C-5408-PFC-M-Rep1:chr5:173609431:T:A\n",
      "ASD-C-5408-Temp-M-Rep2:chr5:173609431:T:A\n",
      "ASD-P-1182-Occi-F-Rep1:chr12:103985722:G:T\n",
      "ASD-P-1182-Temp-F-Rep1:chr12:103985722:G:T\n",
      "['ASD-C-5408-Occi-M-Rep1:chr5:173609431:T:A', 'ASD-C-5408-PFC-M-Rep1:chr5:173609431:T:A', 'ASD-C-5408-Temp-M-Rep2:chr5:173609431:T:A', 'ASD-P-1182-Occi-F-Rep1:chr12:103985722:G:T', 'ASD-P-1182-Temp-F-Rep1:chr12:103985722:G:T']\n"
     ]
    }
   ],
   "source": [
    "f_ans = open(\"../resource/Answer.shared.txt\", 'r')\n",
    "dic_ans = {\"TRUE\":[], 'FALSE':[], \"TRUE_VAF\" : [], \"FALSE_VAF\": []}\n",
    "dic_shared = {'TRUE':{}, 'FALSE':{}}\n",
    "#For set analysis\n",
    "l_TP_set = []\n",
    "l_FP_set = []\n",
    "dic_ans_xVAF = {}\n",
    "\n",
    "for line in f_ans:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if line[0] == \"X\":\n",
    "        l_header = s\n",
    "        print (l_header)\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        #print (dic)\n",
    "        sample = dic[\"Sample\"]\n",
    "        chrom = dic[\"X.CHROM\"]\n",
    "        pos = dic[\"POS\"]\n",
    "        ref = dic[\"REF\"]\n",
    "        alt = dic[\"ALT\"]\n",
    "        VAF = float(dic['VAF_Validation'].rstrip('%')) *0.01\n",
    "\n",
    "        locus = chrom + ':' + pos\n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        dic_ans_xVAF[info] = VAF * 100\n",
    "            \n",
    "        validation = dic[\"Validation\"]\n",
    "        if validation == \"TRUE\":\n",
    "            dic_ans['TRUE'].append(info)\n",
    "            dic_ans['TRUE_VAF'].append(float(VAF))\n",
    "            l_TP_set.append(info)\n",
    "            if locus not in dic_shared[\"TRUE\"]:\n",
    "                dic_shared['TRUE'][locus] = [VAF*100]\n",
    "            else:\n",
    "                dic_shared['TRUE'][locus].append(VAF*100)\n",
    "        else:\n",
    "            dic_ans['FALSE'].append(info)\n",
    "            dic_ans['FALSE_VAF'].append(float(VAF))\n",
    "            l_FP_set.append(info)\n",
    "            if locus not in dic_shared[\"FALSE\"]:\n",
    "                dic_shared['FALSE'][locus] = [VAF*100]\n",
    "            else:\n",
    "                dic_shared['FALSE'][locus].append(VAF*100)\n",
    "\n",
    "for stat in dic_ans:\n",
    "    print (stat, len(dic_ans[stat]))\n",
    "    \n",
    "#### REmove shared that one of them was valiated as false  \n",
    "dic_shared2 = {'TRUE':{}, 'FALSE':{}}\n",
    "for valid in dic_shared:\n",
    "    for locus in dic_shared[valid]:\n",
    "        if len(dic_shared[valid][locus]) != 1:\n",
    "            dic_shared2[valid][locus] = dic_shared[valid][locus]\n",
    "print (\"DIC_SHARED2\", dic_shared2)\n",
    "dic_shared = dic_shared2 \n",
    "    \n",
    "\n",
    "print (dic_ans[\"TRUE\"][0])\n",
    "def Performance_gather(info, VAF, cnt_true, cnt_false, dic_true,dic_false):\n",
    "    variant = info.split(':')[1] + ':' + info.split(':')[2]\n",
    "    if variant in dic_shared[\"TRUE\"]:\n",
    "            cnt_true +=1\n",
    "            if variant not in dic_true:\n",
    "                dic_true[variant] = [VAF]\n",
    "            else:\n",
    "                dic_true[variant].append(VAF)\n",
    "    elif variant in dic_shared[\"FALSE\"]:\n",
    "            cnt_false +=1\n",
    "            if variant not in dic_false:\n",
    "                dic_false[variant] = [VAF]\n",
    "            else:\n",
    "                dic_false[variant].append(VAF)\n",
    "    else:\n",
    "        pass\n",
    "            #print (info, \"Neither in TRUE nor FALSE\")\n",
    "    return [cnt_true, cnt_false, dic_true, dic_false]\n",
    "\n",
    "from scipy.stats import hmean\n",
    "def Performance_check(cnt_true, cnt_false):\n",
    "    TP = cnt_true\n",
    "    FP = cnt_false\n",
    "    PC = len(dic_ans['TRUE'])\n",
    "    NC = len(dic_ans['FALSE'])\n",
    "    sens = TP/PC\n",
    "    if TP == 0 and FP== 0:\n",
    "        pre = None\n",
    "        f_score = None\n",
    "    else:\n",
    "        pre = TP/(TP + FP)\n",
    "        f_score = hmean([sens,pre])\n",
    "    print (sens, pre, f_score)\n",
    "    return (TP, FP, sens, pre, f_score)\n",
    "\n",
    "for e_f in dic_ans['FALSE']:\n",
    "    print (e_f)\n",
    "print (dic_ans['FALSE'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "TRUE Total shared var count:  51\n",
      "FALSE Total shared var count:  5\n"
     ]
    }
   ],
   "source": [
    "## Shared VAF combination\n",
    "l_result = []\n",
    "out = open('/data/project/MRS/BioValidation/PlosGen/analysis/Parse_result.paired.shared.txt', 'w')\n",
    "out.write(\"%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\t%s\\n\" %(\"tool\",\"range\", \"PC\", \"TP\", \"FP\", \"Sensitivity\", \"Precision\", \"F1-score\"))\n",
    "from scipy.stats import hmean\n",
    "cnt_sha = 0\n",
    "for pos in dic_shared['TRUE']:\n",
    "    cnt_sha += len(dic_shared['TRUE'][pos])\n",
    "print (\"TRUE Total shared var count: \", cnt_sha)\n",
    "\n",
    "cnt_sha = 0\n",
    "for pos in dic_shared['FALSE']:\n",
    "    cnt_sha += len(dic_shared['FALSE'][pos])\n",
    "print (\"FALSE Total shared var count: \", cnt_sha)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DIC_PC {'chr1:10659715': [22.07, 21.83], 'chr1:121183580': [8.5, 8.5], 'chr1:152308217': [2.6, 2.9, 2.9, 2.5], 'chr2:111858311': [3.6000000000000005, 3.6000000000000005], 'chr2:219420155': [4.2, 4.2], 'chr5:733811': [6.5, 6.4, 6.4, 6.800000000000001], 'chr5:837373': [5.4, 5.4], 'chr6:136192149': [7.000000000000001, 7.000000000000001], 'chr7:23585592': [24.5, 28.4], 'chr7:55157725': [2.5, 2.5], 'chr9:127676696': [1.8000000000000003, 1.7000000000000002], 'chr12:389107': [8.5, 8.0, 7.6, 8.1, 7.6, 7.8], 'chr12:122701415': [8.4, 8.4], 'chr12:122701470': [8.1, 9.5, 8.1], 'chr15:74722831': [2.3, 2.3], 'chr16:27749710': [7.400000000000001, 7.400000000000001], 'chr16:69463844': [3.3000000000000003, 3.3000000000000003], 'chr17:18781163': [10.6, 10.4], 'chr20:36209646': [4.5, 4.5], 'chr21:44550435': [7.400000000000001, 9.7], 'chr22:38670902': [4.0, 4.0]}\n",
      "DIC_FP {'chr5:173609431': [1.8000000000000003, 1.8000000000000003, 1.8000000000000003], 'chr12:103985722': [5.4, 5.4]}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "{'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "{'chr5:173609431': [1.8000000000000003, 1.8000000000000003, 1.8000000000000003], 'chr12:103985722': [5.4, 5.4]}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "{'5:5': 3, '10:5': 0, '10:10': 1, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n"
     ]
    }
   ],
   "source": [
    "import itertools\n",
    "\n",
    "\n",
    "def Parse_in_comb(dic):\n",
    "    dic_template = {}\n",
    "    Misclassified = 0\n",
    "    l_range = [0,5,10,25,90]\n",
    "    for i in range(len(l_range)-1):\n",
    "        #print (i)\n",
    "        for j in range(len(l_range)-1):\n",
    "            if i>=j:\n",
    "                dic_template[str(l_range[i+1])+':'+str(l_range[j+1])] = 0\n",
    "    for var in dic:\n",
    "        l_vaf = dic[var]\n",
    "        if len(l_vaf) > 1:\n",
    "            l_vaf_comb = (list(itertools.combinations(l_vaf,2)))\n",
    "            #print (\"L_VAF\", l_vaf, l_vaf_comb)\n",
    "\n",
    "            for i in l_vaf_comb:\n",
    "                a = float(i[0]) \n",
    "                b = float(i[1]) \n",
    "                if b > a:\n",
    "                    a,b = b,a\n",
    "                a_stat = None\n",
    "                b_stat = None\n",
    "                for j in range(len(l_range)-1):\n",
    "                    if l_range[j] <= a < l_range[j+1]:\n",
    "                        a_stat = l_range[j+1]\n",
    "                    if l_range[j] <= b < l_range[j+1]:\n",
    "                        b_stat = l_range[j+1]\n",
    "                #print (a, b, a_stat, b_stat)\n",
    "                #if a_stat == None:\n",
    "                 #   a_stat = 90\n",
    "                #if b_stat ==None:\n",
    "                #    b_stat = 90\n",
    "                #print (a, a_stat)\n",
    "                #print (b, b_stat)\n",
    "                vaf_comb = str(a_stat) + ':' + str(b_stat)\n",
    "                #print (vaf_comb)\n",
    "                dic_template[vaf_comb] +=1\n",
    "        elif len(l_vaf) == 1:\n",
    "            Misclassified +=1\n",
    "            print (var, l_vaf)\n",
    "    print (\"MISCLASSIFIED CALLS: \", Misclassified)\n",
    "        \n",
    "    return dic_template\n",
    "\n",
    "def Parse_in_comb_Mis(dic):\n",
    "    dic_Mis = {}\n",
    "    Misclassified = 0\n",
    "    l_range = [0,5,10,25,90]\n",
    "    for i in range(len(l_range)-1):\n",
    "        #print (i)\n",
    "        for j in range(len(l_range)-1):\n",
    "            if i>=j:\n",
    "                dic_template[str(l_range[i+1])+':'+str(l_range[j+1])] = 0\n",
    "    for var in dic:\n",
    "        l_vaf = dic[var]\n",
    "\n",
    "        \n",
    "\n",
    "\n",
    "dic_PC = dic_shared[\"TRUE\"]\n",
    "print (\"DIC_PC\", dic_PC)\n",
    "print (\"DIC_FP\", dic_shared[\"FALSE\"])\n",
    "dic_vafcomb = {}\n",
    "dic_vafcomb = Parse_in_comb(dic_PC)\n",
    "\n",
    "print (dic_vafcomb)\n",
    "\n",
    "def dic_with_VAFcomb(dic):\n",
    "    l_range = [0,5,10,25,90]\n",
    "    for i in range(len(l_range)-1):\n",
    "        #print (i)\n",
    "        for j in range(len(l_range)-1):\n",
    "            if i>=j:\n",
    "                dic[str(l_range[i+1])+':'+str(l_range[j+1])] = 0\n",
    "    return dic\n",
    "\n",
    "print (dic_shared['FALSE'])\n",
    "print (Parse_in_comb(dic_shared['FALSE']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.35294117647058826 1.0 0.5217391304347826\n",
      "(18, 0, 0.35294117647058826, 1.0, 0.5217391304347826)\n",
      "DIC_TRUE: {'chr15:74722831': [2.3, 2.3], 'chr16:69463844': [3.3000000000000003, 3.3000000000000003], 'chr20:36209646': [4.5, 4.5], 'chr9:127676696': [1.8000000000000003, 1.7000000000000002], 'chr1:10659715': [22.07, 21.83], 'chr2:219420155': [4.2, 4.2], 'chr6:136192149': [7.000000000000001, 7.000000000000001], 'chr16:27749710': [7.400000000000001, 7.400000000000001], 'chr22:38670902': [4.0, 4.0]}\n",
      "DIC_FALSE:  {}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "TP {'5:5': 6, '10:5': 0, '10:10': 2, '25:5': 0, '25:10': 0, '25:25': 1, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "10:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "10:10\n",
      "MISCLASSIFIED CALLS:  0\n",
      "25:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "25:10\n",
      "MISCLASSIFIED CALLS:  0\n",
      "25:25\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:10\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:25\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:90\n",
      "MISCLASSIFIED CALLS:  0\n"
     ]
    }
   ],
   "source": [
    "#MF\n",
    "f = open('/data/project/MRS/BioValidation/PlosGen/output/MF/All.snv.Predictions.Mosaic', 'r')\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "dic_TP = dic_with_VAFcomb({})\n",
    "dic_FP = dic_with_VAFcomb({})\n",
    "\n",
    "for line in f:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if s[0] == \"id\":\n",
    "        l_header = s\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        #print (dic)\n",
    "        info_tmp = dic['id']\n",
    "\n",
    "        info_tmp = info_tmp.split('/')[-1]\n",
    "        #print (info_tmp.split('~'))\n",
    "        info = ':'.join(info_tmp.split('~'))\n",
    "        if info in dic_ans_xVAF:\n",
    "            VAF =  dic_ans_xVAF[info]\n",
    "        #print (info) #AD-P-2013029-HIF-F-Rep1:chr8:123182166:A:G\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,dic_true, dic_false)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            dic_true = cnts[2]\n",
    "            dic_false = cnts[3]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['MF', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')\n",
    "    #print (l_result_tmp)\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.6862745098039216 1.0 0.813953488372093\n",
      "(35, 0, 0.6862745098039216, 1.0, 0.813953488372093)\n",
      "DIC_TRUE: {'chr1:152308217': [2.9, 2.5, 2.9], 'chr12:389107': [8.1, 8.0, 7.8, 7.6, 8.5], 'chr12:122701415': [8.4, 8.4], 'chr16:27749710': [7.400000000000001, 7.400000000000001], 'chr9:127676696': [1.7000000000000002, 1.8000000000000003], 'chr1:10659715': [22.07, 21.83], 'chr7:23585592': [24.5, 28.4], 'chr7:55157725': [2.5, 2.5], 'chr16:69463844': [3.3000000000000003, 3.3000000000000003], 'chr20:36209646': [4.5], 'chr22:38670902': [4.0], 'chr17:18781163': [10.6, 10.4], 'chr2:219420155': [4.2, 4.2], 'chr6:136192149': [7.000000000000001, 7.000000000000001], 'chr15:74722831': [2.3, 2.3], 'chr12:122701470': [9.5, 8.1, 8.1]}\n",
      "DIC_FALSE:  {}\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "TP {'5:5': 8, '10:5': 0, '10:10': 16, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:5\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:10\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:5\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:10\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:25\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:5\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:10\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:25\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:90\n",
      "chr20:36209646 [4.5]\n",
      "chr22:38670902 [4.0]\n",
      "MISCLASSIFIED CALLS:  2\n"
     ]
    }
   ],
   "source": [
    "#MT2\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/MT2/*PASS.vcf\")\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            #print (s[8])\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            if info in dic_ans_xVAF:\n",
    "                VAF =  dic_ans_xVAF[info]\n",
    "                cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false,dic_true, dic_false)\n",
    "                cnt_true = cnts[0]\n",
    "                cnt_false = cnts[1]\n",
    "                dic_true = cnts[2]\n",
    "                dic_false = cnts[3]\n",
    "\n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['MT2', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')\n",
    "    #print (l_result_tmp)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0 None None\n",
      "(0, 0, 0.0, None, None)\n",
      "DIC_TRUE: {}\n",
      "DIC_FALSE:  {}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "TP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "10:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "10:10\n",
      "MISCLASSIFIED CALLS:  0\n",
      "25:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "25:10\n",
      "MISCLASSIFIED CALLS:  0\n",
      "25:25\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:5\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:10\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:25\n",
      "MISCLASSIFIED CALLS:  0\n",
      "90:90\n",
      "MISCLASSIFIED CALLS:  0\n"
     ]
    }
   ],
   "source": [
    "#DM\n",
    "f = open('/data/project/MRS/BioValidation/PlosGen/output/DM_output/All.DM.output.mosaic.txt', 'r')\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "\n",
    "for line in f:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    sample = s[0]\n",
    "    chrom = s[2]\n",
    "    pos = s[3]\n",
    "    ref = s[4]\n",
    "    alt = s[5]\n",
    "    #print (VAF) *100\n",
    "    info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "    if info in dic_ans_xVAF:\n",
    "        VAF =  dic_ans_xVAF[info]\n",
    "        cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false, dic_true, dic_false)\n",
    "        cnt_true = cnts[0]\n",
    "        cnt_false = cnts[1]\n",
    "        dic_true = cnts[2]\n",
    "        dic_false = cnts[3]\n",
    "\n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['DM', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.1568627450980392 1.0 0.2711864406779661\n",
      "(8, 0, 0.1568627450980392, 1.0, 0.2711864406779661)\n",
      "DIC_TRUE: {'chr7:23585592': [24.5, 28.4], 'chr17:18781163': [10.6], 'chr1:10659715': [22.07, 21.83], 'chr9:127676696': [1.7000000000000002, 1.8000000000000003], 'chr12:122701470': [9.5]}\n",
      "DIC_FALSE:  {}\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "TP {'5:5': 1, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 1, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:5\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:10\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:5\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:10\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:25\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:5\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:10\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:25\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:90\n",
      "chr17:18781163 [10.6]\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  2\n"
     ]
    }
   ],
   "source": [
    "#HCp20\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/HCp20/*PASS.vcf\")\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            allele = s[9].split(':')[1]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #print (info)\n",
    "            ref = float(allele.split(',')[0])\n",
    "            alt = float(allele.split(',')[1])\n",
    "            if info in dic_ans_xVAF:\n",
    "                VAF =  dic_ans_xVAF[info]\n",
    "                cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false, dic_true, dic_false)\n",
    "                cnt_true = cnts[0]\n",
    "                cnt_false = cnts[1]\n",
    "                dic_true = cnts[2]\n",
    "                dic_false = cnts[3]\n",
    "    \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['HCp20', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.17647058823529413 1.0 0.30000000000000004\n",
      "(9, 0, 0.17647058823529413, 1.0, 0.30000000000000004)\n",
      "DIC_TRUE: {'chr7:23585592': [28.4, 24.5], 'chr17:18781163': [10.6, 10.4], 'chr1:10659715': [21.83, 22.07], 'chr12:122701470': [9.5], 'chr9:127676696': [1.8000000000000003, 1.7000000000000002]}\n",
      "DIC_FALSE:  {}\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "TP {'5:5': 1, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "10:5\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "10:10\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "25:5\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "25:10\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "25:25\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:5\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:10\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:25\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:90\n",
      "chr12:122701470 [9.5]\n",
      "MISCLASSIFIED CALLS:  1\n"
     ]
    }
   ],
   "source": [
    "#HCp200\n",
    "\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/HCp200/*PASS.vcf\")\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            allele = s[9].split(':')[1]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #print (info)\n",
    "            if info in dic_ans_xVAF:\n",
    "                VAF =  dic_ans_xVAF[info]\n",
    "                cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false, dic_true, dic_false)\n",
    "                cnt_true = cnts[0]\n",
    "                cnt_false = cnts[1]\n",
    "                dic_true = cnts[2]\n",
    "                dic_false = cnts[3]\n",
    "    \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['HCp200', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0392156862745098 1.0 0.07547169811320754\n",
      "(2, 0, 0.0392156862745098, 1.0, 0.07547169811320754)\n",
      "DIC_TRUE: {'chr16:27749710': [7.400000000000001], 'chr20:36209646': [4.5]}\n",
      "DIC_FALSE:  {}\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "TP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:10\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:10\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:25\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:10\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:25\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:90\n",
      "chr16:27749710 [7.400000000000001]\n",
      "chr20:36209646 [4.5]\n",
      "MISCLASSIFIED CALLS:  2\n"
     ]
    }
   ],
   "source": [
    "#MH\n",
    "import glob\n",
    "fl = glob.glob('/data/project/MRS/BioValidation/PlosGen/output/MH/*/run/final.passed.tsv')\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    sample = str(file).split('/')[-3]\n",
    "    for line in f:\n",
    "        s = line.strip().split('\\t')\n",
    "        chrom = s[0]\n",
    "        pos = s[1]\n",
    "        ref = s[6]\n",
    "        alt = s[8]\n",
    "       # VAF = float(s[9])*100/int(s[7])\n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        if info in dic_ans_xVAF:\n",
    "            VAF =  dic_ans_xVAF[info]\n",
    "            cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false, dic_true, dic_false)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            dic_true = cnts[2]\n",
    "            dic_false = cnts[3]\n",
    "    \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['MH', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')\n",
    "    \n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Joint genotyping methods"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.47058823529411764 1.0 0.64\n",
      "(24, 0, 0.47058823529411764, 1.0, 0.64)\n",
      "DIC_TRUE: {'chr12:389107': [7.8, 8.5, 8.1, 7.6, 7.6, 8.0], 'chr22:38670902': [4.0, 4.0], 'chr7:55157725': [2.5, 2.5], 'chr2:219420155': [4.2, 4.2], 'chr6:136192149': [7.000000000000001, 7.000000000000001], 'chr20:36209646': [4.5, 4.5], 'chr16:69463844': [3.3000000000000003, 3.3000000000000003], 'chr15:74722831': [2.3, 2.3], 'chr21:44550435': [9.7, 7.400000000000001], 'chr5:837373': [5.4, 5.4]}\n",
      "DIC_FALSE:  {}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "TP {'5:5': 6, '10:5': 0, '10:10': 18, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '5:5', 14, 6, 0, 0.42857142857142855, 1.0, 0.6]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '10:5', 0, 0, 0, None, None, None]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '10:10', 30, 18, 0, 0.6, 1.0, 0.7499999999999999]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '25:5', 0, 0, 0, None, None, None]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '25:10', 0, 0, 0, None, None, None]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '25:25', 2, 0, 0, 0.0, None, None]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '90:5', 0, 0, 0, None, None, None]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '90:10', 0, 0, 0, None, None, None]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '90:25', 1, 0, 0, 0.0, None, None]\n",
      "MISCLASSIFIED CALLS:  0\n",
      "['MT2m', '90:90', 0, 0, 0, None, None, None]\n"
     ]
    }
   ],
   "source": [
    "#MT2-Modified\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/Paired/MT2PM/*normal.vcf\")\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "\n",
    "l_uniq_control = []\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "            control_sample = line.strip().split('\\t')[-2]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            #print (s[8])\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            if info not in l_uniq_control:\n",
    "                if info in dic_ans_xVAF:\n",
    "                    VAF =  dic_ans_xVAF[info]\n",
    "                    cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false,dic_true, dic_false)\n",
    "                    cnt_true = cnts[0]\n",
    "                    cnt_false = cnts[1]\n",
    "                    dic_true = cnts[2]\n",
    "                    dic_false = cnts[3]\n",
    "                    l_uniq_control.append(info)\n",
    "            ##COntrol\n",
    "            info = control_sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            if info not in l_uniq_control:\n",
    "                if info in dic_ans_xVAF:\n",
    "                    VAF =  dic_ans_xVAF[info]\n",
    "                    cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false,dic_true, dic_false)\n",
    "                    cnt_true = cnts[0]\n",
    "                    cnt_false = cnts[1]\n",
    "                    dic_true = cnts[2]\n",
    "                    dic_false = cnts[3]\n",
    "                    l_uniq_control.append(info)\n",
    "\n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    #print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['MT2m', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')\n",
    "    print (l_result_tmp)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GBM-P-0026-Tumor-F-Rep1\n",
      "Joint Mosaic:  1.0\n",
      "Biggest 1.8964438997045733e-31\n",
      "YES MOSAIC 1.0 1.8964438997045733e-31\n",
      "Joint Mosaic:  0.9445827277856884\n",
      "Biggest 0.05501739205107149\n",
      "YES MOSAIC 0.9445827277856884 0.05501739205107149\n",
      "Joint Mosaic:  1.0\n",
      "Biggest 1.4039446945093801e-07\n",
      "YES MOSAIC 1.0 1.4039446945093801e-07\n",
      "ASD-C-5309-Temp-F-Rep1\n",
      "GBM-P-0187-SVZ-M-Rep1\n",
      "SCZ-C-0007-PFC-M-Rep1\n",
      "Joint Mosaic:  1.0\n",
      "Biggest 2.0171119393771587e-32\n",
      "YES MOSAIC 1.0 2.0171119393771587e-32\n",
      "ASD-P-4999-Heart-M-Rep1\n",
      "NLE-P-0150-Brain-M-Rep1\n",
      "ASD-C-5408-Temp-M-Rep2\n",
      "ASD-P-4999-Occi-M-Rep1\n",
      "GBM-P-0396-Recur-M-Rep1\n",
      "Joint Mosaic:  1.0\n",
      "Biggest 6.615154625040279e-29\n",
      "YES MOSAIC 1.0 6.615154625040279e-29\n",
      "Joint Mosaic:  0.003403768383918788\n",
      "Biggest 0.9965979741107903\n",
      "ASD-P-5771-Occi-F-Rep1\n",
      "ASD-C-5408-Occi-M-Rep1\n",
      "SCZ-P-1002-PFC-M-Rep1\n",
      "ASD-P-5144-Cereb-M-Rep1\n",
      "ASD-P-5574-Occi-M-Rep1\n",
      "ASD-C-5408-Heart-M-Rep1\n",
      "SCZ-C-0002-PFC-M-Rep1\n",
      "ASD-C-5408-PFC-M-Rep1\n",
      "ASD-P-1182-Temp-F-Rep1\n",
      "FCD-P-0023-Lhem-M-Rep1\n",
      "FCD-P-0021-RTemOcci-M-Rep1\n",
      "Joint Mosaic:  1.0\n",
      "Biggest 7.030075675121125e-25\n",
      "YES MOSAIC 1.0 7.030075675121125e-25\n",
      "GBM-P-0881-Tumor-M-Rep1\n",
      "Joint Mosaic:  0.0024166847413194612\n",
      "Biggest 0.9975852059697178\n",
      "Joint Mosaic:  0.004833035665088059\n",
      "Biggest 0.9951762431696444\n",
      "Joint Mosaic:  0.008073279719662504\n",
      "Biggest 0.991927658230462\n",
      "Joint Mosaic:  0.006064011273301497\n",
      "Biggest 0.9939396140282909\n",
      "Joint Mosaic:  0.00719796946549077\n",
      "Biggest 0.9927959572270915\n",
      "Joint Mosaic:  0.005993492966359726\n",
      "Biggest 0.9940082753158984\n",
      "ASD-C-5408-Temp-M-Rep2\n",
      "GBM-P-0881-Blood-M-Rep1\n",
      "GBM-P-0026-Tumor-F-Rep2\n",
      "Joint Mosaic:  0.003258742164980566\n",
      "Biggest 0.9967356687205117\n",
      "Joint Mosaic:  0.9157564790173389\n",
      "Biggest 0.08333549127640784\n",
      "YES MOSAIC 0.9157564790173389 0.08333549127640784\n",
      "Joint Mosaic:  0.1289911753483643\n",
      "Biggest 0.8710037002352223\n",
      "ASD-C-5408-Temp-M-Rep1\n",
      "ASD-C-5408-PFC-M-Rep1\n",
      "ASD-P-4999-PFC-M-Rep1\n",
      "GBM-P-0052-Tumor-M-Rep1\n",
      "Joint Mosaic:  0.012312039495821964\n",
      "Biggest 0.9876885064664279\n",
      "ASD-C-5408-Temp-M-Rep1\n",
      "ASD-C-5408-Heart-M-Rep1\n",
      "SCZ-C-1021-PFC-M-Rep2\n",
      "0.0392156862745098 1.0 0.07547169811320754\n",
      "(2, 0, 0.0392156862745098, 1.0, 0.07547169811320754)\n",
      "DIC_TRUE: {'chr7:23585592': [24.5], 'chr1:10659715': [21.83]}\n",
      "DIC_FALSE:  {}\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "TP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:5\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "10:10\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:5\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:10\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "25:25\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:5\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:10\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:25\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n",
      "90:90\n",
      "chr7:23585592 [24.5]\n",
      "chr1:10659715 [21.83]\n",
      "MISCLASSIFIED CALLS:  2\n"
     ]
    }
   ],
   "source": [
    "#MH-Modified\n",
    "import glob\n",
    "fl = glob.glob('/data/project/MRS/BioValidation/PlosGen/output/Paired/MHP/*/final.passed.tsv')\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    sample = str(file).split('/')[-2]\n",
    "    sample = sample.split('_')[1]\n",
    "    print (sample)\n",
    "    for line in f:\n",
    "        s = line.strip().split('\\t')\n",
    "        chrom = s[0]\n",
    "        pos = s[1]\n",
    "        ref = s[6]\n",
    "        alt = s[8]\n",
    "        joint_mosaic = 10**float(s[43])\n",
    "        print (\"Joint Mosaic: \", joint_mosaic)\n",
    "        biggest = 0\n",
    "        for i in range(28,43):\n",
    "            each_joint = 10**float(s[i])\n",
    "            if each_joint > biggest:\n",
    "                biggest = each_joint\n",
    "        print (\"Biggest\", biggest)\n",
    "        if biggest < joint_mosaic and joint_mosaic > 0.05:\n",
    "            print (\"YES MOSAIC\", joint_mosaic, biggest)\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            if info in dic_ans_xVAF:\n",
    "                VAF =  dic_ans_xVAF[info]\n",
    "                cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false, dic_true, dic_false)\n",
    "                cnt_true = cnts[0]\n",
    "                cnt_false = cnts[1]\n",
    "                dic_true = cnts[2]\n",
    "                dic_false = cnts[3]\n",
    "\n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['MHPm', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.0196078431372549 1.0 0.038461538461538464\n",
      "(1, 0, 0.0196078431372549, 1.0, 0.038461538461538464)\n",
      "DIC_TRUE: {'chr16:27749710': [7.400000000000001]}\n",
      "DIC_FALSE:  {}\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "TP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "FP {'5:5': 0, '10:5': 0, '10:10': 0, '25:5': 0, '25:10': 0, '25:25': 0, '90:5': 0, '90:10': 0, '90:25': 0, '90:90': 0}\n",
      "MISCLASSIFIED CALLS:  0\n",
      "PC {'5:5': 14, '10:5': 0, '10:10': 30, '25:5': 0, '25:10': 0, '25:25': 2, '90:5': 0, '90:10': 0, '90:25': 1, '90:90': 0}\n",
      "5:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "10:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "10:10\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "25:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "25:10\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "25:25\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:5\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:10\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:25\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n",
      "90:90\n",
      "chr16:27749710 [7.400000000000001]\n",
      "MISCLASSIFIED CALLS:  1\n"
     ]
    }
   ],
   "source": [
    "#M2S2MH\n",
    "import glob\n",
    "fl = glob.glob('/data/project/MRS/BioValidation/PlosGen/output/Paired/MH_rescue/MH_rescue.txt*')\n",
    "dic_true,dic_false, cnt_true, cnt_false = {}, {},0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        s = line.strip().split('\\t')\n",
    "        sample = s[0]\n",
    "        chrom = s[1]\n",
    "        pos = s[2]\n",
    "        ref = s[3]\n",
    "        alt = s[4]\n",
    "        \n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        if info in dic_ans_xVAF:\n",
    "            VAF =  dic_ans_xVAF[info]\n",
    "            cnts = Performance_gather(info, float(VAF), cnt_true, cnt_false, dic_true, dic_false)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            dic_true = cnts[2]\n",
    "            dic_false = cnts[3]\n",
    "\n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (\"DIC_TRUE:\", dic_true)\n",
    "print (\"DIC_FALSE: \", dic_false)\n",
    "\n",
    "print (\"TP\", Parse_in_comb(dic_true))\n",
    "print (\"FP\", Parse_in_comb(dic_false))\n",
    "\n",
    "dic_vafcomb =  Parse_in_comb(dic_shared['TRUE'])\n",
    "print (\"PC\", dic_vafcomb)\n",
    "\n",
    "\n",
    "for each in dic_vafcomb:\n",
    "    print (each)\n",
    "    PC = int(dic_vafcomb[each])\n",
    "    TP=int(Parse_in_comb(dic_true)[each])\n",
    "    if len(dic_false) == 0:\n",
    "        FP = 0\n",
    "    else:\n",
    "        FP =int(Parse_in_comb(dic_false)[each])\n",
    "    if PC == 0:\n",
    "        sen, pre, fscore = None, None, None\n",
    "    else:\n",
    "        sen = TP/float(PC)\n",
    "        if TP != 0:\n",
    "            \n",
    "            pre = float(TP)/(TP + FP)\n",
    "            fscore = hmean([sen,pre])\n",
    "        else:\n",
    "            pre = None\n",
    "            fscore = None\n",
    "    l_result_tmp = ['M2S2MH', each, PC, TP, FP, sen, pre, fscore]\n",
    "   \n",
    "    out.write('\\t'.join(map(str, l_result_tmp))+ '\\n')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Best block of shared variants\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 547,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['tool', 'range', 'PC', 'TP', 'FP', 'Sensitivity', 'Precision', 'F1-score']\n",
      "dic 0.7272727272727273\n",
      "current 0.6\n",
      "dic 0.6956521739130435\n",
      "current 0.125\n",
      "dic 1.0\n",
      "current 0.6666666666666666\n",
      "dic 0.13333333333333333\n",
      "current 0.7272727272727273\n",
      "dic 0.6666666666666666\n",
      "current 1.0\n",
      "dic 1.0\n",
      "current 1.0\n",
      "Same\n",
      "dic 0.13333333333333333\n",
      "current 0.7272727272727273\n",
      "dic 1.0\n",
      "current 1.0\n",
      "Same\n",
      "dic 1.0\n",
      "current 1.0\n",
      "Same\n",
      "dic 0.6\n",
      "current 0.7272727272727273\n",
      "dic 0.7499999999999999\n",
      "current 0.6956521739130435\n",
      "5:5 MT2\t5:5\t14\t8\t0\t0.5714285714285714\t1.0\t0.7272727272727273\n",
      "\n",
      "10:5 M2S2MH\t10:5\t0\t0\t0\tNone\tNone\tNone\n",
      "\n",
      "10:10 MT2m\t10:10\t30\t18\t0\t0.6\t1.0\t0.7499999999999999\n",
      "\n",
      "25:5 M2S2MH\t25:5\t0\t0\t0\tNone\tNone\tNone\n",
      "\n",
      "25:10 M2S2MH\t25:10\t0\t0\t0\tNone\tNone\tNone\n",
      "\n",
      "25:25 MT2;HCp200\t25:25\t2\t2\t0\t1.0\t1.0\t1.0\n",
      "\n",
      "90:5 M2S2MH\t90:5\t0\t0\t0\tNone\tNone\tNone\n",
      "\n",
      "90:10 M2S2MH\t90:10\t0\t0\t0\tNone\tNone\tNone\n",
      "\n",
      "90:90 M2S2MH\t90:90\t0\t0\t0\tNone\tNone\tNone\n",
      "\n",
      "90:25 MT2;HCp20;HCp200\t90:25\t1\t1\t0\t1.0\t1.0\t1.0\n",
      "\n"
     ]
    }
   ],
   "source": [
    "f = open('/data/project/MRS/BioValidation/PlosGen/analysis/Parse_result.paired.shared.txt', 'r')\n",
    "out = open('/data/project/MRS/BioValidation/PlosGen/analysis/Parse_result.paired.shared.BEST.txt', 'w')\n",
    "dic_best = {}\n",
    "for line in f:\n",
    "    s = line.strip().split('\\t')\n",
    "    Tool = s[0]\n",
    "    if Tool == 'tool':\n",
    "        l_header = s\n",
    "        print (l_header)\n",
    "        out.write(line)\n",
    "    else:\n",
    "        dic= dict(zip(l_header, s))\n",
    "        Range = dic['range']\n",
    "        if dic['PC'] != '0':\n",
    "            if dic['F1-score'] != \"None\":\n",
    "                if Range not in dic_best:\n",
    "                    dic_best[Range] = line\n",
    "                else:\n",
    "                    current = dic_best[Range]\n",
    "                    print (\"dic\", dic['F1-score'])\n",
    "                    current_fscore = current.strip().split('\\t')[-1]\n",
    "                    print ('current', current_fscore)\n",
    "                    if current_fscore != None and dic['F1-score'] != None:\n",
    "                        if float(current_fscore) < float(dic['F1-score']):\n",
    "                            dic_best[Range] = line\n",
    "                        elif float(current_fscore) == float(dic['F1-score']):\n",
    "                            s = line.split('\\t')\n",
    "                            tool = s[0]\n",
    "                            current_tool = current.strip().split('\\t')[0]\n",
    "                            tools = current_tool + ';' + tool\n",
    "                            s[0] = tools\n",
    "                            dic_best[Range] = '\\t'.join(s)\n",
    "                            print (\"Same\")\n",
    "        else:\n",
    "            dic_best[Range] = line\n",
    "            \n",
    "    #print (line)\n",
    "for Range in dic_best:\n",
    "    e_l = dic_best[Range]\n",
    "    PC = e_l.split('\\t')[3]\n",
    "    s = e_l.split('\\t')\n",
    "    if PC == '0':\n",
    "        s[0] = 'NA'\n",
    "        out.write('\\t'.join(s))\n",
    "    else:\n",
    "        out.write('\\t'.join(s))\n",
    "    print (Range, dic_best[Range])"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## RESCUE MF"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "20\n",
      "11\n",
      "{'chr1:10659715': {'GBM-P-0396-Peri-M-Rep1': 22.07, 'GBM-P-0396-Recur-M-Rep1': 21.83}, 'chr1:121183580': {'ASD-C-5309-Occi-F-Rep1': 8.5, 'ASD-C-5309-Temp-F-Rep1': 8.5}, 'chr1:152308217': {'ASD-C-5408-Heart-M-Rep1': 2.6, 'ASD-C-5408-Occi-M-Rep1': 2.9, 'ASD-C-5408-PFC-M-Rep1': 2.9, 'ASD-C-5408-Temp-M-Rep1': 2.5}, 'chr2:111858311': {'GBM-P-0026-SVZ-F-Rep1': 3.6, 'GBM-P-0026-Tumor-F-Rep2': 3.6}, 'chr2:219420155': {'SCZ-C-0002-Liver-M-Rep1': 4.2, 'SCZ-C-0002-PFC-M-Rep1': 4.2}, 'chr5:733811': {'ASD-P-4999-Heart-M-Rep1': 6.5, 'ASD-P-4999-Occi-M-Rep1': 6.4, 'ASD-P-4999-PFC-M-Rep1': 6.4, 'ASD-P-4999-Temp-M-Rep1': 6.8}, 'chr5:837373': {'NLE-P-0150-Blood-M-Rep1': 5.4, 'NLE-P-0150-Brain-M-Rep1': 5.4}, 'chr5:173609431': {'ASD-C-5408-Occi-M-Rep1': 1.8, 'ASD-C-5408-PFC-M-Rep1': 1.8, 'ASD-C-5408-Temp-M-Rep2': 1.8}, 'chr6:136192149': {'SCZ-C-0002-Liver-M-Rep1': 7.0, 'SCZ-C-0002-PFC-M-Rep1': 7.0}, 'chr7:23585592': {'GBM-P-0026-Tumor-F-Rep1': 24.5, 'GBM-P-0026-Tumor-F-Rep2': 28.4}, 'chr7:55157725': {'GBM-P-0052-SVZ-M-Rep1': 2.5, 'GBM-P-0052-Tumor-M-Rep1': 2.5}, 'chr9:127676696': {'GBM-P-0187-Blood-M-Rep1': 1.8, 'GBM-P-0187-SVZ-M-Rep1': 1.7}, 'chr12:389107': {'ASD-C-5408-Cereb-M-Rep1': 8.5, 'ASD-C-5408-Heart-M-Rep1': 8.0, 'ASD-C-5408-Occi-M-Rep1': 7.6, 'ASD-C-5408-PFC-M-Rep1': 8.1, 'ASD-C-5408-Temp-M-Rep1': 7.6, 'ASD-C-5408-Temp-M-Rep2': 7.8}, 'chr12:103985722': {'ASD-P-1182-Occi-F-Rep1': 5.4, 'ASD-P-1182-Temp-F-Rep1': 5.4}, 'chr12:122701415': {'SCZ-C-0007-Liver-M-Rep1': 8.4, 'SCZ-C-0007-PFC-M-Rep1': 8.4}, 'chr12:122701470': {'GBM-P-0881-Blood-M-Rep1': 8.1, 'GBM-P-0881-SVZ-M-Rep1': 9.5, 'GBM-P-0881-Tumor-M-Rep1': 8.1}, 'chr15:74722831': {'ASD-P-5574-Heart-M-Rep1': 2.3, 'ASD-P-5574-Occi-M-Rep1': 2.3}, 'chr16:27749710': {'SCZ-C-1021-Liver-M-Rep1': 7.4, 'SCZ-C-1021-PFC-M-Rep2': 7.4}, 'chr16:69463844': {'FCD-P-0021-Blood-M-Rep1': 3.3, 'FCD-P-0021-RTemOcci-M-Rep1': 3.3}, 'chr17:18781163': {'ASD-P-5771-Cereb-F-Rep1': 10.6, 'ASD-P-5771-Occi-F-Rep1': 10.4}, 'chr20:36209646': {'FCD-P-0023-Blood-M-Rep1': 4.5, 'FCD-P-0023-Lhem-M-Rep1': 4.5}, 'chr21:44550435': {'ASD-P-5144-Cereb-M-Rep1': 7.4, 'ASD-P-5144-Heart-M-Rep1': 9.7}, 'chr22:38670902': {'SCZ-P-1002-Liver-M-Rep1': 4.0, 'SCZ-P-1002-PFC-M-Rep1': 4.0}}\n",
      "{'chr8:123182166': ['AD-P-2013029-HIF-F-Rep1'], 'chr22:40315361': ['AD-P-2013029-HIF-F-Rep1'], 'chr4:99347143': ['AD-P-2013066-HIF-F-Rep1'], 'chr13:42992009': ['AD-P-2013066-HIF-F-Rep1'], 'chr15:98896836': ['AD-P-2013066-HIF-F-Rep1'], 'chr20:17494172': ['AD-P-2013066-HIF-F-Rep1'], 'chr22:46257097': ['AD-P-2013066-HIF-F-Rep1'], 'chr11:67628329': ['AD-P-2014054-HIF-F-Rep1'], 'chr12:4626950': ['ASD-P-1349-Occi-M-Rep1'], 'chr12:113191328': ['ASD-P-1638-PFC-F-Rep1'], 'chr20:46366499': ['ASD-P-4721-Temp-M-Rep1'], 'chr5:179559872': ['ASD-P-4899-Occi-M-Rep1'], 'chr15:74722831': ['ASD-P-5574-Heart-M-Rep1', 'ASD-P-5574-Occi-M-Rep1'], 'chr9:84294263': ['ASD-P-5574-Occi-M-Rep1'], 'chr7:100964707': ['FCD-P-0021-Blood-M-Rep1', 'FCD-P-0021-RTemOcci-M-Rep1', 'SCZ-C-0007-Liver-M-Rep1', 'SCZ-C-0007-PFC-M-Rep1'], 'chr16:69463844': ['FCD-P-0021-Blood-M-Rep1', 'FCD-P-0021-RTemOcci-M-Rep1'], 'chr20:36209646': ['FCD-P-0023-Blood-M-Rep1', 'FCD-P-0023-Lhem-M-Rep1'], 'chr11:128480240': ['GBM-P-0026-Tumor-F-Rep2'], 'chr19:8371354': ['GBM-P-0026-Tumor-F-Rep2'], 'chr21:33262947': ['GBM-P-0026-Tumor-F-Rep2'], 'chr6:87245535': ['GBM-P-0027-Tumor-M-Rep1'], 'chr17:74352100': ['GBM-P-0027-Tumor-M-Rep1'], 'chr17:9969721': ['GBM-P-0052-SVZ-M-Rep2'], 'chr12:89425318': ['GBM-P-0052-Tumor-M-Rep1'], 'chr7:49983464': ['GBM-P-0160-Tumor-M-Rep1'], 'chr15:41844534': ['GBM-P-0160-Tumor-M-Rep1'], 'chr15:99716488': ['GBM-P-0160-Tumor-M-Rep1'], 'chr9:127676696': ['GBM-P-0187-Blood-M-Rep1', 'GBM-P-0187-SVZ-M-Rep1'], 'chr2:210433263': ['GBM-P-0187-SVZ-M-Rep1', 'GBM-P-0187-Tumor-M-Rep1'], 'chr5:113434693': ['GBM-P-0187-Tumor-M-Rep1'], 'chr1:52337866': ['GBM-P-0276-Tumor-M-Rep1'], 'chr2:119248056': ['GBM-P-0276-Tumor-M-Rep1'], 'chr4:86001514': ['GBM-P-0276-Tumor-M-Rep1'], 'chr6:158095820': ['GBM-P-0276-Tumor-M-Rep1'], 'chr11:59422881': ['GBM-P-0276-Tumor-M-Rep1'], 'chr13:78600306': ['GBM-P-0276-Tumor-M-Rep1'], 'chr18:31469150': ['GBM-P-0276-Tumor-M-Rep1'], 'chr21:45909407': ['GBM-P-0276-Tumor-M-Rep1'], 'chr1:10659715': ['GBM-P-0396-Peri-M-Rep1', 'GBM-P-0396-Recur-M-Rep1'], 'chr1:116031514': ['GBM-P-0396-Recur-M-Rep1'], 'chr2:238271458': ['GBM-P-0396-Recur-M-Rep1'], 'chr4:40103074': ['GBM-P-0396-Recur-M-Rep1'], 'chr7:70791458': ['GBM-P-0396-Recur-M-Rep1'], 'chr20:59994312': ['GBM-P-0396-Recur-M-Rep1'], 'chr1:110340663': ['GBM-P-0881-Tumor-M-Rep1'], 'chr2:24039486': ['GBM-P-0881-Tumor-M-Rep1'], 'chr2:218164082': ['GBM-P-0881-Tumor-M-Rep1'], 'chr3:197090990': ['GBM-P-0881-Tumor-M-Rep1'], 'chr11:66332605': ['GBM-P-0881-Tumor-M-Rep1'], 'chr11:129864369': ['GBM-P-0881-Tumor-M-Rep1'], 'chr15:94479623': ['GBM-P-0881-Tumor-M-Rep1'], 'chr16:54112977': ['GBM-P-0881-Tumor-M-Rep1'], 'chr17:7675094': ['GBM-P-0881-Tumor-M-Rep1'], 'chr17:32489134': ['GBM-P-0881-Tumor-M-Rep1'], 'chr5:123180244': ['GG-P-0231-Uncus-M-Rep1'], 'chr1:198742295': ['NLE-P-0225-Brain-M-Rep1'], 'chr1:201047600': ['NLE-P-0225-Brain-M-Rep1'], 'chr1:204450011': ['NLE-P-0225-Brain-M-Rep1'], 'chr1:229595008': ['NLE-P-0225-Brain-M-Rep1'], 'chr1:236569023': ['NLE-P-0225-Brain-M-Rep1'], 'chr19:6583304': ['NLE-P-0227-Brain-M-Rep1'], 'chr2:219420155': ['SCZ-C-0002-Liver-M-Rep1', 'SCZ-C-0002-PFC-M-Rep1'], 'chr6:136192149': ['SCZ-C-0002-Liver-M-Rep1', 'SCZ-C-0002-PFC-M-Rep1'], 'chr2:138015377': ['SCZ-C-1014-PFC-M-Rep2'], 'chr16:27749710': ['SCZ-C-1021-Liver-M-Rep1', 'SCZ-C-1021-PFC-M-Rep2'], 'chr15:73998214': ['SCZ-P-0021-PFC-M-Rep1'], 'chr13:112558343': ['SCZ-P-1001-PFC-M-Rep1'], 'chr22:38670902': ['SCZ-P-1002-Liver-M-Rep1', 'SCZ-P-1002-PFC-M-Rep1'], 'chr12:54966142': ['SCZ-P-1004-PFC-M-Rep1'], 'chrX:107576369': ['AD-P-2013066-HIF-F-Rep1'], 'chrX:154860535': ['GBM-P-0160-Tumor-M-Rep1']}\n",
      "chr1:10659715 GBM-P-0396-Peri-M-Rep1 GBM-P-0396-Recur-M-Rep1 1 1\n",
      "chr1:121183580 ASD-C-5309-Occi-F-Rep1 ASD-C-5309-Temp-F-Rep1 0 0\n",
      "chr1:152308217 ASD-C-5408-Heart-M-Rep1 ASD-C-5408-Occi-M-Rep1 0 0\n",
      "chr1:152308217 ASD-C-5408-Heart-M-Rep1 ASD-C-5408-PFC-M-Rep1 0 0\n",
      "chr1:152308217 ASD-C-5408-Heart-M-Rep1 ASD-C-5408-Temp-M-Rep1 0 0\n",
      "chr1:152308217 ASD-C-5408-Occi-M-Rep1 ASD-C-5408-PFC-M-Rep1 0 0\n",
      "chr1:152308217 ASD-C-5408-Occi-M-Rep1 ASD-C-5408-Temp-M-Rep1 0 0\n",
      "chr1:152308217 ASD-C-5408-PFC-M-Rep1 ASD-C-5408-Temp-M-Rep1 0 0\n",
      "chr2:111858311 GBM-P-0026-SVZ-F-Rep1 GBM-P-0026-Tumor-F-Rep2 0 0\n",
      "chr2:219420155 SCZ-C-0002-Liver-M-Rep1 SCZ-C-0002-PFC-M-Rep1 1 1\n",
      "chr5:733811 ASD-P-4999-Heart-M-Rep1 ASD-P-4999-Occi-M-Rep1 0 0\n",
      "chr5:733811 ASD-P-4999-Heart-M-Rep1 ASD-P-4999-PFC-M-Rep1 0 0\n",
      "chr5:733811 ASD-P-4999-Heart-M-Rep1 ASD-P-4999-Temp-M-Rep1 0 0\n",
      "chr5:733811 ASD-P-4999-Occi-M-Rep1 ASD-P-4999-PFC-M-Rep1 0 0\n",
      "chr5:733811 ASD-P-4999-Occi-M-Rep1 ASD-P-4999-Temp-M-Rep1 0 0\n",
      "chr5:733811 ASD-P-4999-PFC-M-Rep1 ASD-P-4999-Temp-M-Rep1 0 0\n",
      "chr5:837373 NLE-P-0150-Blood-M-Rep1 NLE-P-0150-Brain-M-Rep1 0 0\n",
      "chr5:173609431 ASD-C-5408-Occi-M-Rep1 ASD-C-5408-PFC-M-Rep1 0 0\n",
      "chr5:173609431 ASD-C-5408-Occi-M-Rep1 ASD-C-5408-Temp-M-Rep2 0 0\n",
      "chr5:173609431 ASD-C-5408-PFC-M-Rep1 ASD-C-5408-Temp-M-Rep2 0 0\n",
      "chr6:136192149 SCZ-C-0002-Liver-M-Rep1 SCZ-C-0002-PFC-M-Rep1 1 1\n",
      "chr7:23585592 GBM-P-0026-Tumor-F-Rep1 GBM-P-0026-Tumor-F-Rep2 0 0\n",
      "chr7:55157725 GBM-P-0052-SVZ-M-Rep1 GBM-P-0052-Tumor-M-Rep1 0 0\n",
      "chr9:127676696 GBM-P-0187-Blood-M-Rep1 GBM-P-0187-SVZ-M-Rep1 1 1\n",
      "chr12:389107 ASD-C-5408-Cereb-M-Rep1 ASD-C-5408-Heart-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Cereb-M-Rep1 ASD-C-5408-Occi-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Cereb-M-Rep1 ASD-C-5408-PFC-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Cereb-M-Rep1 ASD-C-5408-Temp-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Cereb-M-Rep1 ASD-C-5408-Temp-M-Rep2 0 0\n",
      "chr12:389107 ASD-C-5408-Heart-M-Rep1 ASD-C-5408-Occi-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Heart-M-Rep1 ASD-C-5408-PFC-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Heart-M-Rep1 ASD-C-5408-Temp-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Heart-M-Rep1 ASD-C-5408-Temp-M-Rep2 0 0\n",
      "chr12:389107 ASD-C-5408-Occi-M-Rep1 ASD-C-5408-PFC-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Occi-M-Rep1 ASD-C-5408-Temp-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-Occi-M-Rep1 ASD-C-5408-Temp-M-Rep2 0 0\n",
      "chr12:389107 ASD-C-5408-PFC-M-Rep1 ASD-C-5408-Temp-M-Rep1 0 0\n",
      "chr12:389107 ASD-C-5408-PFC-M-Rep1 ASD-C-5408-Temp-M-Rep2 0 0\n",
      "chr12:389107 ASD-C-5408-Temp-M-Rep1 ASD-C-5408-Temp-M-Rep2 0 0\n",
      "chr12:103985722 ASD-P-1182-Occi-F-Rep1 ASD-P-1182-Temp-F-Rep1 0 0\n",
      "chr12:122701415 SCZ-C-0007-Liver-M-Rep1 SCZ-C-0007-PFC-M-Rep1 0 0\n",
      "chr12:122701470 GBM-P-0881-Blood-M-Rep1 GBM-P-0881-SVZ-M-Rep1 0 0\n",
      "chr12:122701470 GBM-P-0881-Blood-M-Rep1 GBM-P-0881-Tumor-M-Rep1 0 0\n",
      "chr12:122701470 GBM-P-0881-SVZ-M-Rep1 GBM-P-0881-Tumor-M-Rep1 0 0\n",
      "chr15:74722831 ASD-P-5574-Heart-M-Rep1 ASD-P-5574-Occi-M-Rep1 1 1\n",
      "chr16:27749710 SCZ-C-1021-Liver-M-Rep1 SCZ-C-1021-PFC-M-Rep2 1 1\n",
      "chr16:69463844 FCD-P-0021-Blood-M-Rep1 FCD-P-0021-RTemOcci-M-Rep1 1 1\n",
      "chr17:18781163 ASD-P-5771-Cereb-F-Rep1 ASD-P-5771-Occi-F-Rep1 0 0\n",
      "chr20:36209646 FCD-P-0023-Blood-M-Rep1 FCD-P-0023-Lhem-M-Rep1 1 1\n",
      "chr21:44550435 ASD-P-5144-Cereb-M-Rep1 ASD-P-5144-Heart-M-Rep1 0 0\n",
      "chr22:38670902 SCZ-P-1002-Liver-M-Rep1 SCZ-P-1002-PFC-M-Rep1 1 1\n"
     ]
    }
   ],
   "source": [
    "def get_HQ(chrom,pos,sample):\n",
    "    f = open(\"/data/project/MRS/BioValidation/PlosGen/analysis/Rescue/input/PU.vcf\", 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '#C':\n",
    "            #print (line)\n",
    "            l_header = line.strip().split('\\t')\n",
    "        elif line[0] != '#':\n",
    "            s = line.strip().split('\\t')\n",
    "            dic = dict(zip(l_header, s))\n",
    "            if chrom == dic['#CHROM'] and pos == dic['POS']:\n",
    "                FORMAT = dic[sample]\n",
    "                AD = FORMAT.split(':')\n",
    "                alt = int(AD[1].split(',')[1])\n",
    "    return (alt)\n",
    "            \n",
    "print (get_HQ('chr16', '27749710', 'SCZ-C-1021-Liver-M-Rep1'))\n",
    "print (get_HQ('chr16', '27749710', 'SCZ-C-1021-PFC-M-Rep2'))\n",
    "\n",
    "import glob\n",
    "dic_pair = {}\n",
    "f_ss = open(\"/data/project/MRS/BioValidation/PlosGen/resource/Answer.paired.shared.txt\", 'r')\n",
    "for line in f_ss:\n",
    "    s = line.strip().split('\\t')\n",
    "    if s[0] != 'X.CHROM':\n",
    "        sample = s[6]\n",
    "        s_ID = sample.split('-')\n",
    "        ID = '-'.join(s_ID[:3])\n",
    "        vaf = float(s[5].rstrip(\"%\"))\n",
    "        \n",
    "        #print (ID) #GBM-P-0396\n",
    "        #print (sample)\n",
    "        locus = s[0] + ':' + s[1]\n",
    "        if locus not in dic_pair:\n",
    "            dic_pair[locus] = {sample:vaf}\n",
    "        elif locus in dic_pair:\n",
    "            dic_pair[locus][sample] = vaf\n",
    "print (dic_pair)\n",
    "\n",
    "f_counter = open(\"/data/project/MRS/BioValidation/PlosGen/output/MF/All.snv.Predictions.Mosaic\", 'r')\n",
    "rl_counter = f_counter.readlines()\n",
    "dic_locus_sample = {}\n",
    "for line_counter in rl_counter:\n",
    "    if 'id' not in line_counter:\n",
    "        info_tmp = line_counter.split('\\t')[0]\n",
    "        info_tmp = info_tmp.split('/')[-1]\n",
    "        info = info_tmp.split('~')\n",
    "        tmp_locus = info[1] + ':' + info[2]\n",
    "        tmp_sample = info[0]\n",
    "        if tmp_locus not in dic_locus_sample:\n",
    "            dic_locus_sample[tmp_locus] = [tmp_sample]\n",
    "        else:\n",
    "            dic_locus_sample[tmp_locus].append(tmp_sample)\n",
    "print (dic_locus_sample)\n",
    "            \n",
    "import itertools\n",
    "for locus in dic_pair:\n",
    "    l_sample = dic_pair[locus]\n",
    "    pairs = list(itertools.combinations(l_sample,2))\n",
    "    for each_comb in pairs:\n",
    "        sample_1 = each_comb[0]\n",
    "        sample_2 = each_comb[1]\n",
    "        sample1_stat = 0\n",
    "        sample2_stat = 0\n",
    "        if locus in dic_locus_sample:\n",
    "            if sample_1  in dic_locus_sample[locus]:\n",
    "                sample1_stat = 1\n",
    "            if sample_2 in dic_locus_sample[locus]:\n",
    "                sample2_stat = 1\n",
    "            \n",
    "        print (locus, sample_1, sample_2, sample1_stat, sample2_stat)\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Paired sample analysis - Sample-specific variants parsing\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['X.CHROM', 'POS', 'REF', 'ALT', 'Mappability', 'VAF_Validation', 'Sample', 'Group', 'Validation', 'VAF', 'SET', 'info', 'n']\n",
      "TRUE 25\n",
      "FALSE 3\n",
      "TRUE_VAF 25\n",
      "FALSE_VAF 3\n",
      "GBM-P-0396-Recur-M-Rep1:chr1:116031514:G:A\n"
     ]
    }
   ],
   "source": [
    "f_ans = open(\"../resource/Answer.paired.ss.txt\", 'r')\n",
    "dic_ans = {\"TRUE\":[], 'FALSE':[], \"TRUE_VAF\" : [], \"FALSE_VAF\": []}\n",
    "\n",
    "#For set analysis\n",
    "l_TP_set = []\n",
    "l_FP_set = []\n",
    "\n",
    "\n",
    "for line in f_ans:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if line[0] == \"X\":\n",
    "        l_header = s\n",
    "        print (l_header)\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        #print (dic)\n",
    "        sample = dic[\"Sample\"]\n",
    "        chrom = dic[\"X.CHROM\"]\n",
    "        pos = dic[\"POS\"]\n",
    "        ref = dic[\"REF\"]\n",
    "        alt = dic[\"ALT\"]\n",
    "        VAF = float(dic['VAF_Validation'].rstrip('%')) * 0.01\n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        shared_stat = None\n",
    "\n",
    "        validation = dic[\"Validation\"]\n",
    "        if validation == \"TRUE\":\n",
    "            dic_ans['TRUE'].append(info)\n",
    "            dic_ans['TRUE_VAF'].append(float(VAF))\n",
    "            l_TP_set.append(info)\n",
    "        else:\n",
    "            dic_ans['FALSE'].append(info)\n",
    "            dic_ans['FALSE_VAF'].append(float(VAF))\n",
    "            l_FP_set.append(info)\n",
    "\n",
    "for stat in dic_ans:\n",
    "    print (stat, len(dic_ans[stat]))\n",
    "    \n",
    "\n",
    "print (dic_ans[\"TRUE\"][0])\n",
    "def Performance_gather(info, VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF):\n",
    "    if info in dic_ans[\"TRUE\"]:\n",
    "            cnt_true +=1\n",
    "            l_true.append(info)\n",
    "    elif info in dic_ans[\"FALSE\"]:\n",
    "            cnt_false +=1\n",
    "            l_false.append(info)\n",
    "            l_false_VAF.append(VAF)\n",
    "    else:\n",
    "        pass\n",
    "            #print (info, \"Neither in TRUE nor FALSE\")\n",
    "    return [cnt_true, cnt_false, l_false, l_false_VAF]\n",
    "\n",
    "from scipy.stats import hmean\n",
    "def Performance_check(cnt_true, cnt_false):\n",
    "    TP = cnt_true\n",
    "    FP = cnt_false\n",
    "    PC = len(dic_ans['TRUE'])\n",
    "    NC = len(dic_ans['FALSE'])\n",
    "    sens = TP/PC\n",
    "    if TP == 0 and FP== 0:\n",
    "        pre = None\n",
    "        f_score = None\n",
    "    else:\n",
    "        pre = TP/(TP + FP)\n",
    "        f_score = hmean([sens,pre])\n",
    "    print (sens, pre, f_score)\n",
    "    return (TP, FP, sens, pre, f_score)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 515,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.56 0.875 0.6829268292682927\n",
      "(14, 2, 0.56, 0.875, 0.6829268292682927)\n",
      "['GBM-P-0881-Tumor-M-Rep1:chr2:24039486:C:A', 'GBM-P-0881-Tumor-M-Rep1:chr2:218164082:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr3:197090990:C:T', 'GBM-P-0881-Tumor-M-Rep1:chr11:66332605:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr11:129864369:T:C', 'GBM-P-0881-Tumor-M-Rep1:chr15:94479623:A:T', 'GBM-P-0881-Tumor-M-Rep1:chr16:54112977:G:T', 'GBM-P-0881-Tumor-M-Rep1:chr17:7675094:A:C', 'GBM-P-0881-Tumor-M-Rep1:chr17:32489134:C:A', 'GBM-P-0187-Tumor-M-Rep1:chr5:113434693:G:A', 'GBM-P-0026-Tumor-F-Rep2:chr11:128480240:G:A', 'GBM-P-0026-Tumor-F-Rep2:chr21:33262947:C:A', 'GBM-P-0052-Tumor-M-Rep1:chr12:89425318:A:T', 'GBM-P-0052-SVZ-M-Rep2:chr17:9969721:C:G']\n",
      "['GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T', 'GBM-P-0026-Tumor-F-Rep2:chr19:8371354:G:A']\n",
      "['0.0835913312693498', '0.0607142857142857']\n",
      "0.56 0.875 0.6829268292682927\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "55"
      ]
     },
     "execution_count": 515,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#MF\n",
    "f = open('/data/project/MRS/BioValidation/PlosGen/output/SS/MF/All.ss.tsv', 'r')\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "\n",
    "for line in f:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    if s[0] == \"id\":\n",
    "        l_header = s\n",
    "    else:\n",
    "        dic = dict(zip(l_header, s))\n",
    "        info_tmp = dic['id']\n",
    "        VAF = dic['AF']\n",
    "        info_tmp = info_tmp.split('/')[-1]\n",
    "        #print (info_tmp.split('~'))\n",
    "        info = ':'.join(info_tmp.split('~'))\n",
    "        #print (info) #AD-P-2013029-HIF-F-Rep1:chr8:123182166:A:G\n",
    "        cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "        cnt_true = cnts[0]\n",
    "        cnt_false = cnts[1]\n",
    "        l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_true)\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "\n",
    "\n",
    "l_result_tmp = ['MF']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 516,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "0.52 0.8125 0.6341463414634146\n",
      "(13, 3, 0.52, 0.8125, 0.6341463414634146)\n",
      "['GBM-P-0026-Tumor-F-Rep2:chr19:8371354:G:A', 'GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T']\n",
      "['0.069', '0.279', '0.085']\n",
      "0.52 0.8125 0.6341463414634146\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "57"
      ]
     },
     "execution_count": 516,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#MT2\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/SS/MT2/*tsv\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            sample = str(file).split('/')[-1]\n",
    "            sample = sample.split('.')[0]\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            print (s[8])\n",
    "            VAF = s[9].split(':')[2]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #ref = float(allele.split(',')[0])\n",
    "            #alt = float(allele.split(',')[0])\n",
    "            #VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['MT2']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 517,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "5 0\n",
      "0.2 1.0 0.3333333333333333\n",
      "(5, 0, 0.2, 1.0, 0.3333333333333333)\n",
      "[]\n",
      "[]\n",
      "0.2 1.0 0.3333333333333333\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "51"
      ]
     },
     "execution_count": 517,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#DM\n",
    "f = open('/data/project/MRS/BioValidation/PlosGen/output/SS/DM/All.ss.tsv', 'r')\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "\n",
    "for line in f:\n",
    "    #print (line)\n",
    "    s = line.strip().split('\\t')\n",
    "    sample = s[0]\n",
    "    chrom = s[2]\n",
    "    pos = s[3]\n",
    "    ref = s[4]\n",
    "    alt = s[5]\n",
    "    VAF = s[7]\n",
    "    info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "\n",
    "    cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "    cnt_true = cnts[0]\n",
    "    cnt_false = cnts[1]\n",
    "    l_false = cnts[2]\n",
    "    \n",
    "print (cnt_true, cnt_false)\n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['DM']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 518,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.4 0.9090909090909091 0.5555555555555556\n",
      "(10, 1, 0.4, 0.9090909090909091, 0.5555555555555556)\n",
      "['GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A']\n",
      "[0.27673649393605293]\n",
      "0.4 0.9090909090909091 0.5555555555555556\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "71"
      ]
     },
     "execution_count": 518,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HCp20\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/SS/HCp20/*tsv\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            sample = str(file).split('/')[-1]\n",
    "            sample = sample.split('.')[0]\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            allele = s[9].split(':')[1]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            ref = float(allele.split(',')[0])\n",
    "            alt = float(allele.split(',')[1])\n",
    "            VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['HC-p20']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 519,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.4 0.8333333333333334 0.5405405405405405\n",
      "(10, 2, 0.4, 0.8333333333333334, 0.5405405405405405)\n",
      "['GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T']\n",
      "[0.27673649393605293, 0.08284023668639054]\n",
      "0.4 0.8333333333333334 0.5405405405405405\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "72"
      ]
     },
     "execution_count": 519,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#HCp200\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/SS/HCp200/*tsv\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            s = line.strip().split('\\t')\n",
    "            sample = str(file).split('/')[-1]\n",
    "            sample = sample.split('.')[0]\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            allele = s[9].split(':')[1]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            ref = float(allele.split(',')[0])\n",
    "            alt = float(allele.split(',')[1])\n",
    "            VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['HC-p200']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 520,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0.08 1.0 0.14814814814814814\n",
      "(2, 0, 0.08, 1.0, 0.14814814814814814)\n",
      "[]\n",
      "[]\n",
      "0.08 1.0 0.14814814814814814\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "53"
      ]
     },
     "execution_count": 520,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#MH\n",
    "import glob\n",
    "fl = glob.glob('/data/project/MRS/BioValidation/PlosGen/output/SS/MH/*.tsv')\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    sample = str(file).split('/')[-3]\n",
    "    for line in f:\n",
    "        sample = str(file).split('/')[-1]\n",
    "        sample = sample.split('.')[0]\n",
    "        s = line.strip().split('\\t')\n",
    "        chrom = s[0]\n",
    "        pos = s[1]\n",
    "        ref = s[6]\n",
    "        alt = s[8]\n",
    "        VAF = float(s[9])/int(s[7])\n",
    "        info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "        cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "        cnt_true = cnts[0]\n",
    "        cnt_false = cnts[1]\n",
    "        l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "        \n",
    "l_result_tmp = ['MH']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp )) + '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Joint"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 521,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "GT:AD:AF:DP:F1R2:F2R1:FAD:SB\n",
      "0.76 0.9047619047619048 0.8260869565217391\n",
      "(19, 2, 0.76, 0.9047619047619048, 0.8260869565217391)\n",
      "['GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T']\n",
      "['9.500e-03', '2.899e-03']\n",
      "0.76 0.9047619047619048 0.8260869565217391\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "70"
      ]
     },
     "execution_count": 521,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#MT2P\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/SS/MT2P/*tsv\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            sample = str(file).split('/')[-1]\n",
    "            sample = sample.split('.')[0]\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            print (s[8])\n",
    "            VAF = s[9].split(':')[2]\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #ref = float(allele.split(',')[0])\n",
    "            #alt = float(allele.split(',')[0])\n",
    "            #VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['MT2P']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 522,
   "metadata": {},
   "outputs": [],
   "source": [
    "def STK(line):\n",
    "        dic_idx = {'A': 0, 'C': 1, 'G':2, 'T':3}\n",
    "        if line[0] != '#':\n",
    "               s = line.strip().split('\\t')\n",
    "               format = s[-1]\n",
    "               ref = s[3]\n",
    "               alt = s[4]\n",
    "               l_format = format.split(':')[4:]\n",
    "               dp_cnt = int(format.split(':')[0])\n",
    "               alt_cnt= int(l_format[dic_idx[alt]].split(',')[0])\n",
    "               vaf = float(alt_cnt)/dp_cnt\n",
    "               return vaf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 523,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "0.68 0.85 0.7555555555555555\n",
      "(17, 3, 0.68, 0.85, 0.7555555555555555)\n",
      "['GBM-P-0026-Tumor-F-Rep2:chr19:8371354:G:A', 'GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T']\n",
      "[0.06031746031746032, 0.2763440860215054, 0.08357348703170028]\n",
      "0.68 0.85 0.7555555555555555\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "55"
      ]
     },
     "execution_count": 523,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#STK\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/SS/STK/*tsv\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            sample = str(file).split('/')[-1]\n",
    "            sample = sample.split('.')[0]\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            print (s[8])\n",
    "            VAF = STK(line)\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #ref = float(allele.split(',')[0])\n",
    "            #alt = float(allele.split(',')[0])\n",
    "            #VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['STK']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 524,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "DP:FDP:SDP:SUBDP:AU:CU:GU:TU\n",
      "0.48 0.8571428571428571 0.6153846153846154\n",
      "(12, 2, 0.48, 0.8571428571428571, 0.6153846153846154)\n",
      "['GBM-P-0187-Tumor-M-Rep1:chr20:17456366:G:A', 'GBM-P-0881-Tumor-M-Rep1:chr1:110340663:A:T']\n",
      "[0.2763440860215054, 0.08357348703170028]\n",
      "0.48 0.8571428571428571 0.6153846153846154\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "72"
      ]
     },
     "execution_count": 524,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#M2S2MH\n",
    "import glob\n",
    "fl = glob.glob(\"/data/project/MRS/BioValidation/PlosGen/output/SS/M2S2MH/*tsv\")\n",
    "l_true, l_false,l_false_VAF, cnt_true, cnt_false = [], [], [],0,0\n",
    "for file in fl:\n",
    "    f = open(file, 'r')\n",
    "    for line in f:\n",
    "        if line[:2] == '##':\n",
    "            pass\n",
    "        elif line[0] == '#':\n",
    "            sample = line.strip().split('\\t')[-1]\n",
    "        else:\n",
    "            sample = str(file).split('/')[-1]\n",
    "            sample = sample.split('.')[0]\n",
    "            s = line.strip().split('\\t')\n",
    "            chrom = s[0]\n",
    "            pos = s[1]\n",
    "            ref = s[3]\n",
    "            alt = s[4]\n",
    "            print (s[8])\n",
    "            VAF = STK(line)\n",
    "            info = sample + ':' + chrom + ':' + pos + ':' + ref + ':' + alt\n",
    "            #ref = float(allele.split(',')[0])\n",
    "            #alt = float(allele.split(',')[0])\n",
    "            #VAF = alt/(ref+alt)\n",
    "            cnts = Performance_gather(info,VAF, cnt_true, cnt_false,l_true, l_false, l_false_VAF)\n",
    "            cnt_true = cnts[0]\n",
    "            cnt_false = cnts[1]\n",
    "            l_false = cnts[2]\n",
    "        \n",
    "print (Performance_check(cnt_true, cnt_false))\n",
    "print (l_false)\n",
    "print (l_false_VAF)\n",
    "\n",
    "l_result_tmp = ['M2S2MH']\n",
    "l_result_tmp.extend(list(Performance_check(cnt_true, cnt_false)))\n",
    "out.write('\\t'.join(map(str,l_result_tmp ))+ '\\t'+ DATA + '\\t' + VAF_range + '\\n')\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
